{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler, MinMaxScaler\n",
    "from sklearn.model_selection import train_test_split, GridSearchCV\n",
    "from sklearn.linear_model import LinearRegression, Lasso, Ridge, ElasticNet\n",
    "from sklearn.metrics import mean_squared_error\n",
    "from sklearn.datasets import load_breast_cancer, fetch_california_housing, load_diabetes\n",
    "from sklearn.feature_selection import SelectFromModel\n",
    "from sklearn.ensemble import RandomForestRegressor\n",
    "from sklearn.pipeline import Pipeline\n",
    "from scipy.stats import spearmanr, pearsonr\n",
    "\n",
    "import numpy as np\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.model_selection import GridSearchCV, train_test_split\n",
    "from sklearn.ensemble import RandomForestRegressor\n",
    "from sklearn.metrics import r2_score\n",
    "import matplotlib.cm as cm\n",
    "\n",
    "from sklearn.manifold import TSNE\n",
    "import pandas as pd\n",
    "import itertools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%config InlineBackend.figure_format='retina'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_17640\\2358411230.py:2: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256,257,258,259,260,261,262,263,264,265,266,267,268,269,270,271,272,273,274,275,276,277,278,279,280,281,282,283,284,285,286,287,288,289,290,291,292) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df_orthogroups = pd.read_csv(orthoG, sep='\\t')\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[3], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m orthoG \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mC:/Users/46705/Documents/SpiderSilk/data/annotations/Orthogroups_ls.tsv\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m----> 2\u001b[0m df_orthogroups \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[43morthoG\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;130;43;01m\\t\u001b[39;49;00m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m \n\u001b[0;32m      3\u001b[0m first_last_columns \u001b[38;5;241m=\u001b[39m df_orthogroups\u001b[38;5;241m.\u001b[39miloc[:, [\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]]\n\u001b[0;32m      5\u001b[0m data_without_na \u001b[38;5;241m=\u001b[39m first_last_columns\u001b[38;5;241m.\u001b[39mdropna(subset\u001b[38;5;241m=\u001b[39m[first_last_columns\u001b[38;5;241m.\u001b[39mcolumns[\u001b[38;5;241m-\u001b[39m\u001b[38;5;241m1\u001b[39m]])\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\io\\parsers\\readers.py:948\u001b[0m, in \u001b[0;36mread_csv\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[0;32m    935\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[0;32m    936\u001b[0m     dialect,\n\u001b[0;32m    937\u001b[0m     delimiter,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    944\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[0;32m    945\u001b[0m )\n\u001b[0;32m    946\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[1;32m--> 948\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\io\\parsers\\readers.py:617\u001b[0m, in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    614\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n\u001b[0;32m    616\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m parser:\n\u001b[1;32m--> 617\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mparser\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnrows\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\io\\parsers\\readers.py:1765\u001b[0m, in \u001b[0;36mTextFileReader.read\u001b[1;34m(self, nrows)\u001b[0m\n\u001b[0;32m   1762\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m   1763\u001b[0m         new_rows \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlen\u001b[39m(index)\n\u001b[1;32m-> 1765\u001b[0m     df \u001b[38;5;241m=\u001b[39m \u001b[43mDataFrame\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcol_dict\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mindex\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1767\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_currow \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m new_rows\n\u001b[0;32m   1768\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m df\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\core\\frame.py:733\u001b[0m, in \u001b[0;36mDataFrame.__init__\u001b[1;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[0;32m    727\u001b[0m     mgr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_init_mgr(\n\u001b[0;32m    728\u001b[0m         data, axes\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mindex\u001b[39m\u001b[38;5;124m\"\u001b[39m: index, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcolumns\u001b[39m\u001b[38;5;124m\"\u001b[39m: columns}, dtype\u001b[38;5;241m=\u001b[39mdtype, copy\u001b[38;5;241m=\u001b[39mcopy\n\u001b[0;32m    729\u001b[0m     )\n\u001b[0;32m    731\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data, \u001b[38;5;28mdict\u001b[39m):\n\u001b[0;32m    732\u001b[0m     \u001b[38;5;66;03m# GH#38939 de facto copy defaults to False only in non-dict cases\u001b[39;00m\n\u001b[1;32m--> 733\u001b[0m     mgr \u001b[38;5;241m=\u001b[39m \u001b[43mdict_to_mgr\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtyp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmanager\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    734\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data, ma\u001b[38;5;241m.\u001b[39mMaskedArray):\n\u001b[0;32m    735\u001b[0m     \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnumpy\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mma\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m mrecords\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\core\\internals\\construction.py:503\u001b[0m, in \u001b[0;36mdict_to_mgr\u001b[1;34m(data, index, columns, dtype, typ, copy)\u001b[0m\n\u001b[0;32m    499\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    500\u001b[0m         \u001b[38;5;66;03m# dtype check to exclude e.g. range objects, scalars\u001b[39;00m\n\u001b[0;32m    501\u001b[0m         arrays \u001b[38;5;241m=\u001b[39m [x\u001b[38;5;241m.\u001b[39mcopy() \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(x, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdtype\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01melse\u001b[39;00m x \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m arrays]\n\u001b[1;32m--> 503\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43marrays_to_mgr\u001b[49m\u001b[43m(\u001b[49m\u001b[43marrays\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtyp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtyp\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mconsolidate\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\core\\internals\\construction.py:152\u001b[0m, in \u001b[0;36marrays_to_mgr\u001b[1;34m(arrays, columns, index, dtype, verify_integrity, typ, consolidate)\u001b[0m\n\u001b[0;32m    149\u001b[0m axes \u001b[38;5;241m=\u001b[39m [columns, index]\n\u001b[0;32m    151\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m typ \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mblock\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m--> 152\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mcreate_block_manager_from_column_arrays\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m    153\u001b[0m \u001b[43m        \u001b[49m\u001b[43marrays\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxes\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mconsolidate\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mconsolidate\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrefs\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrefs\u001b[49m\n\u001b[0;32m    154\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    155\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m typ \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124marray\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[0;32m    156\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m ArrayManager(arrays, [index, columns])\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\core\\internals\\managers.py:2086\u001b[0m, in \u001b[0;36mcreate_block_manager_from_column_arrays\u001b[1;34m(arrays, axes, consolidate, refs)\u001b[0m\n\u001b[0;32m   2068\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mcreate_block_manager_from_column_arrays\u001b[39m(\n\u001b[0;32m   2069\u001b[0m     arrays: \u001b[38;5;28mlist\u001b[39m[ArrayLike],\n\u001b[0;32m   2070\u001b[0m     axes: \u001b[38;5;28mlist\u001b[39m[Index],\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   2082\u001b[0m     \u001b[38;5;66;03m# These last three are sufficient to allow us to safely pass\u001b[39;00m\n\u001b[0;32m   2083\u001b[0m     \u001b[38;5;66;03m#  verify_integrity=False below.\u001b[39;00m\n\u001b[0;32m   2085\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m-> 2086\u001b[0m         blocks \u001b[38;5;241m=\u001b[39m \u001b[43m_form_blocks\u001b[49m\u001b[43m(\u001b[49m\u001b[43marrays\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mconsolidate\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrefs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   2087\u001b[0m         mgr \u001b[38;5;241m=\u001b[39m BlockManager(blocks, axes, verify_integrity\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)\n\u001b[0;32m   2088\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\core\\internals\\managers.py:2160\u001b[0m, in \u001b[0;36m_form_blocks\u001b[1;34m(arrays, consolidate, refs)\u001b[0m\n\u001b[0;32m   2157\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28missubclass\u001b[39m(dtype\u001b[38;5;241m.\u001b[39mtype, (\u001b[38;5;28mstr\u001b[39m, \u001b[38;5;28mbytes\u001b[39m)):\n\u001b[0;32m   2158\u001b[0m     dtype \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mdtype(\u001b[38;5;28mobject\u001b[39m)\n\u001b[1;32m-> 2160\u001b[0m values, placement \u001b[38;5;241m=\u001b[39m \u001b[43m_stack_arrays\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mlist\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mtup_block\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   2161\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_dtlike:\n\u001b[0;32m   2162\u001b[0m     values \u001b[38;5;241m=\u001b[39m ensure_wrapped_if_datetimelike(values)\n",
      "File \u001b[1;32mc:\\Users\\46705\\miniconda3\\Lib\\site-packages\\pandas\\core\\internals\\managers.py:2202\u001b[0m, in \u001b[0;36m_stack_arrays\u001b[1;34m(tuples, dtype)\u001b[0m\n\u001b[0;32m   2200\u001b[0m stacked \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mempty(shape, dtype\u001b[38;5;241m=\u001b[39mdtype)\n\u001b[0;32m   2201\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i, arr \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28menumerate\u001b[39m(arrays):\n\u001b[1;32m-> 2202\u001b[0m     stacked[i] \u001b[38;5;241m=\u001b[39m arr\n\u001b[0;32m   2204\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m stacked, placement\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "orthoG = \"C:/Users/46705/Documents/SpiderSilk/data/annotations/Orthogroups_ls.tsv\"\n",
    "df_orthogroups = pd.read_csv(orthoG, sep='\\t') \n",
    "first_last_columns = df_orthogroups.iloc[:, [0, -1]]\n",
    "\n",
    "data_without_na = first_last_columns.dropna(subset=[first_last_columns.columns[-1]])\n",
    "data_without_na\n",
    "data_exploded = data_without_na.assign(last_column=data_without_na[data_without_na.columns[-1]].str.split(\", \")).explode('last_column')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "path_lodings_mag_ara_sum =  \"C:/Users/46705/Documents/SpiderSilk/output/loadings/Ara_MaG_sum.csv\"\n",
    "path_lodings_mag_ara_count =  \"C:/Users/46705/Documents/SpiderSilk/output/loadings/ara_GC_PC3.csv\"\n",
    "path_lodings_mag_ara_bool_1 =  \"C:/Users/46705/Documents/SpiderSilk/output/loadings/MaG_bool_PC1.csv\"\n",
    "path_lodings_mag_ara_bool_2 = \"C:/Users/46705/Documents/SpiderSilk/output/loadings/MaG_bool_PC2.csv\"\n",
    "# path_lodings_all_sum =  \"C:/Users/46705/Documents/SpiderSilk/output/loadings/all_sum.csv\"\n",
    "# path_lodings_ara_sum =  \"C:/Users/46705/Documents/SpiderSilk/output/loadings/ara_sum_PC2.csv\"\n",
    "\n",
    "ara_mag_sum = pd.read_csv(path_lodings_mag_ara_sum, sep = \",\")\n",
    "ara_mag_GC = pd.read_csv(path_lodings_mag_ara_count, sep = \",\")\n",
    "# all_sum = pd.read_csv(path_lodings_all_sum, sep = \",\")\n",
    "# ara_sum = pd.read_csv(path_lodings_ara_sum, sep = \",\")\n",
    "mag_ara_bool_1 = pd.read_csv(path_lodings_mag_ara_bool_1, sep = \",\")\n",
    "mag_ara_bool_2 = pd.read_csv(path_lodings_mag_ara_bool_2, sep = \",\")\n",
    "\n",
    "proteomics_gland = \"C:/Users/46705/Documents/SpiderSilk/data/proteomics/proteinsInGland.2024.csv\"\n",
    "prot_gland = pd.read_csv(proteomics_gland, sep = \",\")\n",
    "\n",
    "proteomics_silk = \"C:/Users/46705/Documents/SpiderSilk/data/proteomics/proteinsInSilk.2024.csv\"\n",
    "prot_silk = pd.read_csv(proteomics_silk, sep = \",\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OG_id</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>Group</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Product</th>\n",
       "      <th>Lascid</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OG0000264</td>\n",
       "      <td>0.222516</td>\n",
       "      <td>-0.075873</td>\n",
       "      <td>0.128765</td>\n",
       "      <td>-0.161316</td>\n",
       "      <td>-0.055178</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC001444,LASC001445,LASC001446,LASC001447,LA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       OG_id       PC1       PC2       PC3       PC4       PC5  Group  \\\n",
       "4  OG0000264  0.222516 -0.075873  0.128765 -0.161316 -0.055178  SpiCE   \n",
       "\n",
       "                                            GeneName               Product  \\\n",
       "4  Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...  hypothetical protein   \n",
       "\n",
       "                                              Lascid cluster  \n",
       "4  LASC001444,LASC001445,LASC001446,LASC001447,LA...     NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ara_mag_sum[ara_mag_sum[\"OG_id\"] == \"OG0000264\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'OG_id' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[14], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m lista_top10 \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(ara_mag_sum[\u001b[43mOG_id\u001b[49m]\u001b[38;5;241m.\u001b[39mhead(\u001b[38;5;241m10\u001b[39m))\n",
      "\u001b[1;31mNameError\u001b[0m: name 'OG_id' is not defined"
     ]
    }
   ],
   "source": [
    "lista_top10 = list(ara_mag_sum[OG_id].head(10))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## add OG to silkset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'data_exploded' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[6], line 3\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;66;03m# silksetWog = prot_silk[[\"geneName\", \"product\", \"lascID\", \"scGeneName\", \"silkGroup\"]]\u001b[39;00m\n\u001b[0;32m      2\u001b[0m silksetWog \u001b[38;5;241m=\u001b[39m prot_silk\n\u001b[1;32m----> 3\u001b[0m merged_df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mmerge(silksetWog, \u001b[43mdata_exploded\u001b[49m[[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mlast_column\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mOrthogroup\u001b[39m\u001b[38;5;124m\"\u001b[39m]], how\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mleft\u001b[39m\u001b[38;5;124m'\u001b[39m, left_on\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mlascID\u001b[39m\u001b[38;5;124m'\u001b[39m, right_on\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mlast_column\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m      5\u001b[0m \u001b[38;5;66;03m# Drop the 'lascID' column (optional)\u001b[39;00m\n\u001b[0;32m      6\u001b[0m merged_df\u001b[38;5;241m.\u001b[39mdrop(columns\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mlast_column\u001b[39m\u001b[38;5;124m'\u001b[39m, inplace\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'data_exploded' is not defined"
     ]
    }
   ],
   "source": [
    "# silksetWog = prot_silk[[\"geneName\", \"product\", \"lascID\", \"scGeneName\", \"silkGroup\"]]\n",
    "silksetWog = prot_silk\n",
    "merged_df = pd.merge(silksetWog, data_exploded[[\"last_column\", \"Orthogroup\"]], how='left', left_on='lascID', right_on='last_column')\n",
    "\n",
    "merged_df.drop(columns='last_column', inplace=True)\n",
    "silk_og = merged_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneNameProtein</th>\n",
       "      <th>scGeneName</th>\n",
       "      <th>Orthogroup</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SpiCE-LMa1</td>\n",
       "      <td>LASC013229</td>\n",
       "      <td>OG0009245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SpiCE-LMa2</td>\n",
       "      <td>xync</td>\n",
       "      <td>OG0013923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SpiCE-LMa3</td>\n",
       "      <td>LASC000319</td>\n",
       "      <td>OG0000171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SpiCE-LMa4</td>\n",
       "      <td>tpo-pxt</td>\n",
       "      <td>OG0000494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SpiCE-LMa6</td>\n",
       "      <td>LASC009843</td>\n",
       "      <td>OG0006927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SpiCE-LMa5</td>\n",
       "      <td>Spice-8</td>\n",
       "      <td>OG0000264</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   geneNameProtein  scGeneName Orthogroup\n",
       "10      SpiCE-LMa1  LASC013229  OG0009245\n",
       "11      SpiCE-LMa2        xync  OG0013923\n",
       "13      SpiCE-LMa3  LASC000319  OG0000171\n",
       "19      SpiCE-LMa4     tpo-pxt  OG0000494\n",
       "22      SpiCE-LMa6  LASC009843  OG0006927\n",
       "23      SpiCE-LMa5     Spice-8  OG0000264"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# silk_og.to_csv(\"C:/Users/46705/Documents/SpiderSilk/data/prot_silk_OG.csv\", index = False)\n",
    "spice_OG =silk_og[silk_og[\"silkGroup\"]== \"Spice-LMa\"][[\"geneNameProtein\", \"scGeneName\", \"Orthogroup\"]][\"Orthogroup\"]\n",
    "# silk_og.to_csv(\"C:/Users/46705/Documents/SpiderSilk/data/prot_silk_OG.csv\", index = False)\n",
    "silk_og[silk_og[\"silkGroup\"]== \"Spice-LMa\"][[\"geneNameProtein\", \"scGeneName\", \"Orthogroup\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4     OG0000264\n",
       "44    OG0006927\n",
       "79    OG0000171\n",
       "Name: OG_id, dtype: object"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "spice_OG_values = set(spice_OG)\n",
    "\n",
    "matching_values = ara_mag_sum[ara_mag_sum[\"OG_id\"].isin(spice_OG_values)][\"OG_id\"]\n",
    "\n",
    "matching_values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "150\n",
      "Number of unique values in silk_og['Orthogroup'] also present in ara_mag_sum['OG_id']: 20\n"
     ]
    }
   ],
   "source": [
    "# silk_og[\"Orthogroup\"].unique() == ara_mag_sum[\"OG_id\"]\n",
    "\n",
    "\n",
    "\n",
    "silk_og_unique = set(silk_og[\"Orthogroup\"].unique())\n",
    "print(len(silk_og_unique))\n",
    "ara_mag_sum_og_ids = set(ara_mag_sum[\"OG_id\"])\n",
    "common_values = silk_og_unique.intersection(ara_mag_sum_og_ids)\n",
    "common_values_count = len(common_values)\n",
    "\n",
    "print(\"Number of unique values in silk_og['Orthogroup'] also present in ara_mag_sum['OG_id']:\", common_values_count)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Odds ratio: 3.6944444444444446\n",
      "p-value: 0.16911955929831424\n"
     ]
    }
   ],
   "source": [
    "from scipy.stats import fisher_exact\n",
    "\n",
    "observed = [[2, 4], [20-2, 173-40]]  \n",
    "odds_ratio, p_value = fisher_exact(observed)\n",
    "\n",
    "print(\"Odds ratio:\", odds_ratio)\n",
    "print(\"p-value:\", p_value)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['OG0000282',\n",
       " 'OG0008243',\n",
       " 'OG0008607',\n",
       " 'OG0008312',\n",
       " 'OG0000264',\n",
       " 'OG0004717',\n",
       " 'OG0007747',\n",
       " 'OG0007860',\n",
       " 'OG0004409',\n",
       " 'OG0000189']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(ara_mag_sum[\"OG_id\"].head(10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OG_id</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>Group</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Product</th>\n",
       "      <th>Lascid</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OG0000282</td>\n",
       "      <td>-0.480160</td>\n",
       "      <td>0.200225</td>\n",
       "      <td>-0.166053</td>\n",
       "      <td>-0.072299</td>\n",
       "      <td>0.005558</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC013610, cd74_2, LASC013607, tg_5, LASC0136...</td>\n",
       "      <td>hypothetical protein,Thyroglobulin,Nidogen-2,H...</td>\n",
       "      <td>LASC013595,LASC013596,LASC013597,LASC013602,LA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OG0008243</td>\n",
       "      <td>-0.398308</td>\n",
       "      <td>0.032694</td>\n",
       "      <td>0.283644</td>\n",
       "      <td>-0.469987</td>\n",
       "      <td>-0.127325</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp-1339#prd34368.1_4</td>\n",
       "      <td>Sp-1339#PRD34368.1 protein</td>\n",
       "      <td>LASC010015</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OG0008607</td>\n",
       "      <td>0.232044</td>\n",
       "      <td>0.343374</td>\n",
       "      <td>-0.385207</td>\n",
       "      <td>-0.163741</td>\n",
       "      <td>-0.549446</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC012976</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC012976</td>\n",
       "      <td>Duct_1, Duct_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OG0008312</td>\n",
       "      <td>-0.227529</td>\n",
       "      <td>-0.017233</td>\n",
       "      <td>-0.135446</td>\n",
       "      <td>0.112192</td>\n",
       "      <td>-0.006052</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC018595</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC018595</td>\n",
       "      <td>ZoneC_SpiCE-LMa1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OG0000264</td>\n",
       "      <td>0.222516</td>\n",
       "      <td>-0.075873</td>\n",
       "      <td>0.128765</td>\n",
       "      <td>-0.161316</td>\n",
       "      <td>-0.055178</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC001444,LASC001445,LASC001446,LASC001447,LA...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>OG0007089</td>\n",
       "      <td>0.001582</td>\n",
       "      <td>-0.057665</td>\n",
       "      <td>-0.027281</td>\n",
       "      <td>-0.023315</td>\n",
       "      <td>-0.007352</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC003517</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC003517</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>OG0006013</td>\n",
       "      <td>0.001425</td>\n",
       "      <td>-0.093077</td>\n",
       "      <td>-0.039545</td>\n",
       "      <td>-0.020985</td>\n",
       "      <td>0.062993</td>\n",
       "      <td>NaN</td>\n",
       "      <td>timm23</td>\n",
       "      <td>Mitochondrial import inner membrane translocas...</td>\n",
       "      <td>LASC017275</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>OG0004738</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>-0.178953</td>\n",
       "      <td>-0.016423</td>\n",
       "      <td>0.064668</td>\n",
       "      <td>-0.095063</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC004484</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC004484</td>\n",
       "      <td>Duct_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>OG0003918</td>\n",
       "      <td>-0.000696</td>\n",
       "      <td>-0.020372</td>\n",
       "      <td>-0.012682</td>\n",
       "      <td>-0.039044</td>\n",
       "      <td>0.082980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rpl19</td>\n",
       "      <td>60S ribosomal protein L19</td>\n",
       "      <td>LASC009569</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>OG0001157</td>\n",
       "      <td>-0.000052</td>\n",
       "      <td>-0.012483</td>\n",
       "      <td>0.022892</td>\n",
       "      <td>0.077352</td>\n",
       "      <td>0.002406</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tmem68, tmem68_2</td>\n",
       "      <td>Transmembrane protein 68</td>\n",
       "      <td>LASC006763,LASC006765</td>\n",
       "      <td>ZoneC_SpiCE-LMa1, nan</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>173 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         OG_id       PC1       PC2       PC3       PC4       PC5  Group  \\\n",
       "0    OG0000282 -0.480160  0.200225 -0.166053 -0.072299  0.005558    NaN   \n",
       "1    OG0008243 -0.398308  0.032694  0.283644 -0.469987 -0.127325    NaN   \n",
       "2    OG0008607  0.232044  0.343374 -0.385207 -0.163741 -0.549446    NaN   \n",
       "3    OG0008312 -0.227529 -0.017233 -0.135446  0.112192 -0.006052    NaN   \n",
       "4    OG0000264  0.222516 -0.075873  0.128765 -0.161316 -0.055178  SpiCE   \n",
       "..         ...       ...       ...       ...       ...       ...    ...   \n",
       "168  OG0007089  0.001582 -0.057665 -0.027281 -0.023315 -0.007352    NaN   \n",
       "169  OG0006013  0.001425 -0.093077 -0.039545 -0.020985  0.062993    NaN   \n",
       "170  OG0004738  0.000700 -0.178953 -0.016423  0.064668 -0.095063    NaN   \n",
       "171  OG0003918 -0.000696 -0.020372 -0.012682 -0.039044  0.082980    NaN   \n",
       "172  OG0001157 -0.000052 -0.012483  0.022892  0.077352  0.002406    NaN   \n",
       "\n",
       "                                              GeneName  \\\n",
       "0    LASC013610, cd74_2, LASC013607, tg_5, LASC0136...   \n",
       "1                                 sp-1339#prd34368.1_4   \n",
       "2                                           LASC012976   \n",
       "3                                           LASC018595   \n",
       "4    Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...   \n",
       "..                                                 ...   \n",
       "168                                         LASC003517   \n",
       "169                                             timm23   \n",
       "170                                         LASC004484   \n",
       "171                                              rpl19   \n",
       "172                                   tmem68, tmem68_2   \n",
       "\n",
       "                                               Product  \\\n",
       "0    hypothetical protein,Thyroglobulin,Nidogen-2,H...   \n",
       "1                           Sp-1339#PRD34368.1 protein   \n",
       "2                                 hypothetical protein   \n",
       "3                                 hypothetical protein   \n",
       "4                                 hypothetical protein   \n",
       "..                                                 ...   \n",
       "168                               hypothetical protein   \n",
       "169  Mitochondrial import inner membrane translocas...   \n",
       "170                               hypothetical protein   \n",
       "171                          60S ribosomal protein L19   \n",
       "172                           Transmembrane protein 68   \n",
       "\n",
       "                                                Lascid                cluster  \n",
       "0    LASC013595,LASC013596,LASC013597,LASC013602,LA...                    NaN  \n",
       "1                                           LASC010015                    NaN  \n",
       "2                                           LASC012976         Duct_1, Duct_2  \n",
       "3                                           LASC018595       ZoneC_SpiCE-LMa1  \n",
       "4    LASC001444,LASC001445,LASC001446,LASC001447,LA...                    NaN  \n",
       "..                                                 ...                    ...  \n",
       "168                                         LASC003517                    NaN  \n",
       "169                                         LASC017275                    NaN  \n",
       "170                                         LASC004484                 Duct_2  \n",
       "171                                         LASC009569                    NaN  \n",
       "172                              LASC006763,LASC006765  ZoneC_SpiCE-LMa1, nan  \n",
       "\n",
       "[173 rows x 11 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ara_mag_sum"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## check how many of 20 OG can be found in silkset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OG_id</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>Group_x</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Product</th>\n",
       "      <th>Lascid</th>\n",
       "      <th>...</th>\n",
       "      <th>lascID</th>\n",
       "      <th>geneName_GOI</th>\n",
       "      <th>Nephila_genome_Class</th>\n",
       "      <th>Group_y</th>\n",
       "      <th>scGeneName</th>\n",
       "      <th>sc_gene_id</th>\n",
       "      <th>Length</th>\n",
       "      <th>geneNameProtein</th>\n",
       "      <th>silkGroup</th>\n",
       "      <th>Orthogroup</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OG0000264</td>\n",
       "      <td>0.222516</td>\n",
       "      <td>-0.075873</td>\n",
       "      <td>0.128765</td>\n",
       "      <td>-0.161316</td>\n",
       "      <td>-0.055178</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC001444,LASC001445,LASC001446,LASC001447,LA...</td>\n",
       "      <td>...</td>\n",
       "      <td>LASC007774</td>\n",
       "      <td>Spice_8</td>\n",
       "      <td>Spice_NMa4</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice-8</td>\n",
       "      <td>maker-000078F-est-gff-StringTie-gene-26.2</td>\n",
       "      <td>147.0</td>\n",
       "      <td>SpiCE-LMa5</td>\n",
       "      <td>Spice-LMa</td>\n",
       "      <td>OG0000264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OG0000264</td>\n",
       "      <td>0.222516</td>\n",
       "      <td>-0.075873</td>\n",
       "      <td>0.128765</td>\n",
       "      <td>-0.161316</td>\n",
       "      <td>-0.055178</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC001444,LASC001445,LASC001446,LASC001447,LA...</td>\n",
       "      <td>...</td>\n",
       "      <td>LASC014000</td>\n",
       "      <td>Spice_15</td>\n",
       "      <td>Spice_NMa4</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice-15</td>\n",
       "      <td>maker-000197F-est-gff-StringTie-gene-10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OG0000264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OG0000189</td>\n",
       "      <td>-0.136838</td>\n",
       "      <td>0.049264</td>\n",
       "      <td>-0.065472</td>\n",
       "      <td>0.100008</td>\n",
       "      <td>0.003191</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pxd_pxt_3, pxt_2, pxd_5, pxd_pxt_4, pxd_pxt, p...</td>\n",
       "      <td>hypothetical protein,Chorion peroxidase,Peroxi...</td>\n",
       "      <td>LASC015555,LASC015557,LASC015558,LASC015560,LA...</td>\n",
       "      <td>...</td>\n",
       "      <td>LASC015562</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pxd-4</td>\n",
       "      <td>maker-000238F-augustus-gene-13.41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OG0000189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OG0000369</td>\n",
       "      <td>-0.120817</td>\n",
       "      <td>0.050641</td>\n",
       "      <td>-0.029833</td>\n",
       "      <td>0.124924</td>\n",
       "      <td>-0.036854</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gld_15, LASC008285, gld_3, gld_19, beta_3</td>\n",
       "      <td>Glucose dehydrogenase [FAD%2C quinone],hypothe...</td>\n",
       "      <td>LASC001976,LASC008285,LASC008287,LASC020658,LA...</td>\n",
       "      <td>...</td>\n",
       "      <td>LASC008287</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>beta-3</td>\n",
       "      <td>evm-000086F-processed-gene-4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OG0000369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OG0002778</td>\n",
       "      <td>-0.097265</td>\n",
       "      <td>-0.012688</td>\n",
       "      <td>-0.018355</td>\n",
       "      <td>0.121373</td>\n",
       "      <td>-0.026275</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>Spidroin3</td>\n",
       "      <td>Uncharacterized, protein, (Fragment)</td>\n",
       "      <td>LASC020272</td>\n",
       "      <td>...</td>\n",
       "      <td>LASC020272</td>\n",
       "      <td>Spidroin3</td>\n",
       "      <td>Unclassified</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>Spidroin3</td>\n",
       "      <td>maker-000428F-est-gff-StringTie-gene-4.8</td>\n",
       "      <td>2350.0</td>\n",
       "      <td>AmSp-like2</td>\n",
       "      <td>AmSp-like</td>\n",
       "      <td>OG0002778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>OG0007367</td>\n",
       "      <td>-0.093221</td>\n",
       "      <td>-0.113635</td>\n",
       "      <td>-0.170399</td>\n",
       "      <td>-0.185069</td>\n",
       "      <td>0.222619</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sec61b</td>\n",
       "      <td>Protein transport protein Sec61 subunit beta</td>\n",
       "      <td>LASC021718</td>\n",
       "      <td>...</td>\n",
       "      <td>LASC021718</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sec61b</td>\n",
       "      <td>maker-000540F-augustus-gene-2.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OG0007367</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       OG_id       PC1       PC2       PC3       PC4       PC5   Group_x  \\\n",
       "0  OG0000264  0.222516 -0.075873  0.128765 -0.161316 -0.055178     SpiCE   \n",
       "1  OG0000264  0.222516 -0.075873  0.128765 -0.161316 -0.055178     SpiCE   \n",
       "2  OG0000189 -0.136838  0.049264 -0.065472  0.100008  0.003191       NaN   \n",
       "3  OG0000369 -0.120817  0.050641 -0.029833  0.124924 -0.036854       NaN   \n",
       "4  OG0002778 -0.097265 -0.012688 -0.018355  0.121373 -0.026275  Spidroin   \n",
       "5  OG0007367 -0.093221 -0.113635 -0.170399 -0.185069  0.222619       NaN   \n",
       "\n",
       "                                            GeneName  \\\n",
       "0  Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...   \n",
       "1  Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...   \n",
       "2  pxd_pxt_3, pxt_2, pxd_5, pxd_pxt_4, pxd_pxt, p...   \n",
       "3          gld_15, LASC008285, gld_3, gld_19, beta_3   \n",
       "4                                          Spidroin3   \n",
       "5                                             sec61b   \n",
       "\n",
       "                                             Product  \\\n",
       "0                               hypothetical protein   \n",
       "1                               hypothetical protein   \n",
       "2  hypothetical protein,Chorion peroxidase,Peroxi...   \n",
       "3  Glucose dehydrogenase [FAD%2C quinone],hypothe...   \n",
       "4               Uncharacterized, protein, (Fragment)   \n",
       "5       Protein transport protein Sec61 subunit beta   \n",
       "\n",
       "                                              Lascid  ...      lascID  \\\n",
       "0  LASC001444,LASC001445,LASC001446,LASC001447,LA...  ...  LASC007774   \n",
       "1  LASC001444,LASC001445,LASC001446,LASC001447,LA...  ...  LASC014000   \n",
       "2  LASC015555,LASC015557,LASC015558,LASC015560,LA...  ...  LASC015562   \n",
       "3  LASC001976,LASC008285,LASC008287,LASC020658,LA...  ...  LASC008287   \n",
       "4                                         LASC020272  ...  LASC020272   \n",
       "5                                         LASC021718  ...  LASC021718   \n",
       "\n",
       "  geneName_GOI  Nephila_genome_Class   Group_y  scGeneName  \\\n",
       "0      Spice_8            Spice_NMa4     SpiCE     Spice-8   \n",
       "1     Spice_15            Spice_NMa4     SpiCE    Spice-15   \n",
       "2          NaN                   NaN       NaN       pxd-4   \n",
       "3          NaN                   NaN       NaN      beta-3   \n",
       "4    Spidroin3          Unclassified  Spidroin   Spidroin3   \n",
       "5          NaN                   NaN       NaN      sec61b   \n",
       "\n",
       "                                  sc_gene_id  Length  geneNameProtein  \\\n",
       "0  maker-000078F-est-gff-StringTie-gene-26.2   147.0       SpiCE-LMa5   \n",
       "1  maker-000197F-est-gff-StringTie-gene-10.0     NaN              NaN   \n",
       "2          maker-000238F-augustus-gene-13.41     NaN              NaN   \n",
       "3             evm-000086F-processed-gene-4.0     NaN              NaN   \n",
       "4   maker-000428F-est-gff-StringTie-gene-4.8  2350.0       AmSp-like2   \n",
       "5            maker-000540F-augustus-gene-2.1     NaN              NaN   \n",
       "\n",
       "   silkGroup Orthogroup  \n",
       "0  Spice-LMa  OG0000264  \n",
       "1        NaN  OG0000264  \n",
       "2        NaN  OG0000189  \n",
       "3        NaN  OG0000369  \n",
       "4  AmSp-like  OG0002778  \n",
       "5        NaN  OG0007367  \n",
       "\n",
       "[6 rows x 31 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common = pd.merge(ara_mag_sum.head(20), silk_og, left_on =\"OG_id\", right_on =\"Orthogroup\" , how = \"inner\")\n",
    "common"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Group</th>\n",
       "      <th>geneName</th>\n",
       "      <th>product</th>\n",
       "      <th>lascID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>OG0000086</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>LASC011985,Spice_18,awa1,tnxb_7,LASC011991,LAS...</td>\n",
       "      <td>hypothetical protein,Cell wall protein AWA1,Te...</td>\n",
       "      <td>LASC011985,LASC011987,LASC011989,LASC011990,LA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>OG0000264</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_3,Spice_4,Spice_5,Spice_6,Spice_2,Spice_...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC001444,LASC001445,LASC001446,LASC001447,LA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1269</th>\n",
       "      <td>OG0001273</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>LASC001401,LASC001402,LASC001457,Spice_17</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC001401,LASC001402,LASC001457,LASC019594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2194</th>\n",
       "      <td>OG0002201</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_19</td>\n",
       "      <td>Tyrosine-protein kinase receptor Tie-1,Multipl...</td>\n",
       "      <td>LASC017498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165894</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp4,MaSp2a,MaSp2b,MaSp2c,MaSp1a,MaSp1b,MaSp1...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165895</th>\n",
       "      <td>OG0000139</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>AcSp1b,AcSp1c</td>\n",
       "      <td>hypothetical, protein</td>\n",
       "      <td>LASC012314,LASC012315</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165896</th>\n",
       "      <td>OG0000344</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>TuSpa,TuSpb</td>\n",
       "      <td>Tubuliform, spidroin, 1, variant, 1, (Fragment)</td>\n",
       "      <td>LASC002331,LASC008470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165897</th>\n",
       "      <td>OG0000846</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>AgSp2a,AgSp2b,agsp-c#prd26655.1</td>\n",
       "      <td>AgSp-b#PRD22063.1 protein,AgSp-a#PRD23399.1 pr...</td>\n",
       "      <td>LASC004210,LASC004211,LASC004212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165898</th>\n",
       "      <td>OG0001354</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>AgSp1</td>\n",
       "      <td>AgSp-c#PRD26655.1 protein</td>\n",
       "      <td>LASC014453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165899</th>\n",
       "      <td>OG0001803</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>FlSp-like</td>\n",
       "      <td>Sp-5803#PRD29580.1 protein</td>\n",
       "      <td>LASC000543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165900</th>\n",
       "      <td>OG0001853</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MiSpd</td>\n",
       "      <td>Minor, ampullate, spidroin, 1, variant, 2, (Fr...</td>\n",
       "      <td>LASC022495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165901</th>\n",
       "      <td>OG0002663</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>FlSpa,FlSpb,FlSpc</td>\n",
       "      <td>Flagelliform spidroin variant 2 (Fragment),Fla...</td>\n",
       "      <td>LASC007997,LASC013942,LASC013944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165902</th>\n",
       "      <td>OG0002778</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>Spidroin3</td>\n",
       "      <td>Uncharacterized, protein, (Fragment)</td>\n",
       "      <td>LASC020272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165903</th>\n",
       "      <td>OG0003877</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>AcSpa</td>\n",
       "      <td>Aciniform, spidroin, 1, variant, 3, (Fragment)</td>\n",
       "      <td>LASC004415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165904</th>\n",
       "      <td>OG0004997</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>PySpa,PySpb</td>\n",
       "      <td>Pyriform, spidroin, 1, variant, 1, (Fragment)</td>\n",
       "      <td>LASC011790,LASC011791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165905</th>\n",
       "      <td>OG0007111</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>Spidroin2</td>\n",
       "      <td>Sp-8175#PRD27848.1 protein</td>\n",
       "      <td>LASC004216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165906</th>\n",
       "      <td>OG0007640</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>Spidroin1</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC000723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165907</th>\n",
       "      <td>OG0008237</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>TuSpc</td>\n",
       "      <td>Tubuliform spidroin 1 variant 1 (Fragment)</td>\n",
       "      <td>LASC022086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165908</th>\n",
       "      <td>OG0009283</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>TuSp-like</td>\n",
       "      <td>hypothetical, protein</td>\n",
       "      <td>LASC001804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165909</th>\n",
       "      <td>OG0010900</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MiSpa</td>\n",
       "      <td>hypothetical, protein</td>\n",
       "      <td>LASC007254</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0     Group  \\\n",
       "85      OG0000086     SpiCE   \n",
       "262     OG0000264     SpiCE   \n",
       "1269    OG0001273     SpiCE   \n",
       "2194    OG0002201     SpiCE   \n",
       "165894  OG0000050  Spidroin   \n",
       "165895  OG0000139  Spidroin   \n",
       "165896  OG0000344  Spidroin   \n",
       "165897  OG0000846  Spidroin   \n",
       "165898  OG0001354  Spidroin   \n",
       "165899  OG0001803  Spidroin   \n",
       "165900  OG0001853  Spidroin   \n",
       "165901  OG0002663  Spidroin   \n",
       "165902  OG0002778  Spidroin   \n",
       "165903  OG0003877  Spidroin   \n",
       "165904  OG0004997  Spidroin   \n",
       "165905  OG0007111  Spidroin   \n",
       "165906  OG0007640  Spidroin   \n",
       "165907  OG0008237  Spidroin   \n",
       "165908  OG0009283  Spidroin   \n",
       "165909  OG0010900  Spidroin   \n",
       "\n",
       "                                                 geneName  \\\n",
       "85      LASC011985,Spice_18,awa1,tnxb_7,LASC011991,LAS...   \n",
       "262     Spice_3,Spice_4,Spice_5,Spice_6,Spice_2,Spice_...   \n",
       "1269            LASC001401,LASC001402,LASC001457,Spice_17   \n",
       "2194                                             Spice_19   \n",
       "165894  MaSp4,MaSp2a,MaSp2b,MaSp2c,MaSp1a,MaSp1b,MaSp1...   \n",
       "165895                                      AcSp1b,AcSp1c   \n",
       "165896                                        TuSpa,TuSpb   \n",
       "165897                    AgSp2a,AgSp2b,agsp-c#prd26655.1   \n",
       "165898                                              AgSp1   \n",
       "165899                                          FlSp-like   \n",
       "165900                                              MiSpd   \n",
       "165901                                  FlSpa,FlSpb,FlSpc   \n",
       "165902                                          Spidroin3   \n",
       "165903                                              AcSpa   \n",
       "165904                                        PySpa,PySpb   \n",
       "165905                                          Spidroin2   \n",
       "165906                                          Spidroin1   \n",
       "165907                                              TuSpc   \n",
       "165908                                          TuSp-like   \n",
       "165909                                              MiSpa   \n",
       "\n",
       "                                                  product  \\\n",
       "85      hypothetical protein,Cell wall protein AWA1,Te...   \n",
       "262                                  hypothetical protein   \n",
       "1269                                 hypothetical protein   \n",
       "2194    Tyrosine-protein kinase receptor Tie-1,Multipl...   \n",
       "165894  MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "165895                              hypothetical, protein   \n",
       "165896    Tubuliform, spidroin, 1, variant, 1, (Fragment)   \n",
       "165897  AgSp-b#PRD22063.1 protein,AgSp-a#PRD23399.1 pr...   \n",
       "165898                          AgSp-c#PRD26655.1 protein   \n",
       "165899                         Sp-5803#PRD29580.1 protein   \n",
       "165900  Minor, ampullate, spidroin, 1, variant, 2, (Fr...   \n",
       "165901  Flagelliform spidroin variant 2 (Fragment),Fla...   \n",
       "165902               Uncharacterized, protein, (Fragment)   \n",
       "165903     Aciniform, spidroin, 1, variant, 3, (Fragment)   \n",
       "165904      Pyriform, spidroin, 1, variant, 1, (Fragment)   \n",
       "165905                         Sp-8175#PRD27848.1 protein   \n",
       "165906                               hypothetical protein   \n",
       "165907         Tubuliform spidroin 1 variant 1 (Fragment)   \n",
       "165908                              hypothetical, protein   \n",
       "165909                              hypothetical, protein   \n",
       "\n",
       "                                                   lascID  \n",
       "85      LASC011985,LASC011987,LASC011989,LASC011990,LA...  \n",
       "262     LASC001444,LASC001445,LASC001446,LASC001447,LA...  \n",
       "1269          LASC001401,LASC001402,LASC001457,LASC019594  \n",
       "2194                                           LASC017498  \n",
       "165894  LASC004263,LASC004265,LASC004267,LASC004269,LA...  \n",
       "165895                              LASC012314,LASC012315  \n",
       "165896                              LASC002331,LASC008470  \n",
       "165897                   LASC004210,LASC004211,LASC004212  \n",
       "165898                                         LASC014453  \n",
       "165899                                         LASC000543  \n",
       "165900                                         LASC022495  \n",
       "165901                   LASC007997,LASC013942,LASC013944  \n",
       "165902                                         LASC020272  \n",
       "165903                                         LASC004415  \n",
       "165904                              LASC011790,LASC011791  \n",
       "165905                                         LASC004216  \n",
       "165906                                         LASC000723  \n",
       "165907                                         LASC022086  \n",
       "165908                                         LASC001804  \n",
       "165909                                         LASC007254  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "# path_tail_sac = \"C:/Users/46705/Documents/SpiderSilk/data/post_filtering/tail_sac_log_multisp.csv\"\n",
    "path_max = \"C:/Users/46705/Documents/SpiderSilk/data/post_filtering/max_tail_sac_log_multisp.csv\"\n",
    "## Full dataset\n",
    "# path = \"C:/Users/46705/Documents/SpiderSilk/data/post_filtering/filtered_multisp_log.csv\"\n",
    "\n",
    "\n",
    "# path = \"C:/Users/46705/Documents/SpiderSilk/data/post_filtering/filtered_full_multisp_log_20_lascID.csv\"\n",
    "# path = \"C:/Users/46705/Documents/SpiderSilk/generalized/filtered_sp_added_90p_nonzero_values_log_multsp.csv\"\n",
    "path_anno = \"C:/Users/46705/Documents/SpiderSilk/data/annotations/annotation.csv\"\n",
    "# path_not_log = \"C:/Users/46705/Documents/SpiderSilk/generalized/filtered_sp_added_90p_nonzero_values.csv\"\n",
    "# df = pd.read_csv(path, sep= \",\")\n",
    "\n",
    "\n",
    "path_MaG = \"C:/Users/46705/Documents/SpiderSilk/data/post_filtering/MaG_ara.csv\"\n",
    "\n",
    "mag =pd.read_csv(path_MaG, sep= \",\")\n",
    "\n",
    "file_connectoion = \"C:/Users/46705/Documents/SpiderSilk/data/raw_data/S1-S4/data_s1.csv\"\n",
    "\n",
    "df_anno = pd.read_csv(path_anno, sep= \",\")\n",
    "\n",
    "spidroin_conts_path = \"C:/Users/46705/Documents/SpiderSilk/data/pre_filtering/df_grouped.csv\"\n",
    "\n",
    "path_anno_ = \"C:/Users/46705/Documents/SpiderSilk/data/annotations/OG_annotations.csv\"\n",
    "\n",
    "anno__ =pd.read_csv(path_anno_, sep= \",\")\n",
    "\n",
    "\n",
    "anno_spidroins = anno__[anno__['Group'].isin([\"Spidroin\", \"SpiCE\"])]\n",
    "anno_spidroins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Decide on what df to look at. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Agreeing OG of high loadings across datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_list = [ara_mag_sum, mag_ara_bool_1, mag_ara_bool_2, ara_mag_GC]\n",
    "df_names = ['ara_mag_sum','mag_ara_bool_1', 'mag_ara_bool_2', \"ara_mag_GC\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ara_mag_sum</th>\n",
       "      <th>mag_ara_bool_1</th>\n",
       "      <th>mag_ara_bool_2</th>\n",
       "      <th>ara_mag_GC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ara_mag_sum</th>\n",
       "      <td></td>\n",
       "      <td>[OG0000282, OG0008243, OG0008607, OG0008312, O...</td>\n",
       "      <td>[OG0000282, OG0008243, OG0008312, OG0007860, O...</td>\n",
       "      <td>[OG0000282, OG0000264, OG0000369, OG0000442, O...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mag_ara_bool_1</th>\n",
       "      <td>[OG0000282, OG0008243, OG0008607, OG0008312, O...</td>\n",
       "      <td></td>\n",
       "      <td>[OG0002778, OG0008312, OG0008243, OG0002319, O...</td>\n",
       "      <td>[OG0000282]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mag_ara_bool_2</th>\n",
       "      <td>[OG0000282, OG0008243, OG0008312, OG0007860, O...</td>\n",
       "      <td>[OG0002778, OG0008312, OG0008243, OG0002319, O...</td>\n",
       "      <td></td>\n",
       "      <td>[OG0000282]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ara_mag_GC</th>\n",
       "      <td>[OG0000282, OG0000264, OG0000369, OG0000442, O...</td>\n",
       "      <td>[OG0000282]</td>\n",
       "      <td>[OG0000282]</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                      ara_mag_sum  \\\n",
       "ara_mag_sum                                                         \n",
       "mag_ara_bool_1  [OG0000282, OG0008243, OG0008607, OG0008312, O...   \n",
       "mag_ara_bool_2  [OG0000282, OG0008243, OG0008312, OG0007860, O...   \n",
       "ara_mag_GC      [OG0000282, OG0000264, OG0000369, OG0000442, O...   \n",
       "\n",
       "                                                   mag_ara_bool_1  \\\n",
       "ara_mag_sum     [OG0000282, OG0008243, OG0008607, OG0008312, O...   \n",
       "mag_ara_bool_1                                                      \n",
       "mag_ara_bool_2  [OG0002778, OG0008312, OG0008243, OG0002319, O...   \n",
       "ara_mag_GC                                            [OG0000282]   \n",
       "\n",
       "                                                   mag_ara_bool_2  \\\n",
       "ara_mag_sum     [OG0000282, OG0008243, OG0008312, OG0007860, O...   \n",
       "mag_ara_bool_1  [OG0002778, OG0008312, OG0008243, OG0002319, O...   \n",
       "mag_ara_bool_2                                                      \n",
       "ara_mag_GC                                            [OG0000282]   \n",
       "\n",
       "                                                       ara_mag_GC  \n",
       "ara_mag_sum     [OG0000282, OG0000264, OG0000369, OG0000442, O...  \n",
       "mag_ara_bool_1                                        [OG0000282]  \n",
       "mag_ara_bool_2                                        [OG0000282]  \n",
       "ara_mag_GC                                                         "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# # Sample data\n",
    "# df_list = [ara_mag_sum, ara_sum, all_sum, mag_ara_bool_1, mag_ara_bool_2]\n",
    "# df_names = ['ara_mag_sum', 'ara_sum', 'all_sum', 'mag_ara_bool_1', 'mag_ara_bool_2']\n",
    "\n",
    "def find_common_og_ids(df1, df2):\n",
    "    merged_df = pd.merge(df1.head(20), df2.head(20), on='OG_id', how='inner')\n",
    "    return list(merged_df[\"OG_id\"])\n",
    "\n",
    "common_og_ids_dict = {}\n",
    "\n",
    "combos = list(itertools.combinations(range(len(df_list)), 2))\n",
    "for combo in combos:\n",
    "    df1_index, df2_index = combo\n",
    "    df1 = df_list[df1_index]\n",
    "    df2 = df_list[df2_index]\n",
    "    common_og_ids = find_common_og_ids(df1, df2)\n",
    "    common_og_ids_dict[(df_names[df1_index], df_names[df2_index])] = common_og_ids\n",
    "\n",
    "matrix = pd.DataFrame(\"\", index=df_names, columns=df_names)\n",
    "\n",
    "for combo, og_ids in common_og_ids_dict.items():\n",
    "    df1_name, df2_name = combo\n",
    "    matrix.loc[df1_name, df2_name] = og_ids\n",
    "    matrix.loc[df2_name, df1_name] = og_ids \n",
    "\n",
    "matrix\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ara_mag_sum</th>\n",
       "      <th>mag_ara_bool_1</th>\n",
       "      <th>mag_ara_bool_2</th>\n",
       "      <th>ara_mag_GC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ara_mag_sum</th>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mag_ara_bool_1</th>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mag_ara_bool_2</th>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ara_mag_GC</th>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                ara_mag_sum  mag_ara_bool_1  mag_ara_bool_2  ara_mag_GC\n",
       "ara_mag_sum               0               7               7           5\n",
       "mag_ara_bool_1            7               0              13           1\n",
       "mag_ara_bool_2            7              13               0           1\n",
       "ara_mag_GC                5               1               1           0"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "matrix_counts = pd.DataFrame(0, index=df_names, columns=df_names)\n",
    "\n",
    "for combo, og_ids in common_og_ids_dict.items():\n",
    "    df1_name, df2_name = combo\n",
    "    count = len(og_ids)\n",
    "    matrix_counts.loc[df1_name, df2_name] = count\n",
    "    matrix_counts.loc[df2_name, df1_name] = count\n",
    "\n",
    "matrix_counts\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_values = list(prot_gland['geneNameProtein'].dropna().unique())\n",
    "prot18 = prot_gland[prot_gland['geneNameProtein'].isin(unique_values)]\n",
    "prot18 = pd.merge(prot18, prot_silk[[\"lascID\", \"geneNameProtein\"]], on= \"geneNameProtein\", how= \"left\")\n",
    "# anno_exploded = anno__['lascID'].str.split(',').explode().astype(str)\n",
    "anno_exploded = anno__.assign(lascID=anno__['lascID'].str.split(','))\n",
    "anno_exploded = anno_exploded.explode('lascID')\n",
    "anno_exploded[anno_exploded[\"lascID\"]== \"LASC007774\"]\n",
    "\n",
    "\n",
    "merged_df = pd.merge(prot18, anno_exploded, how='left', on='lascID')\n",
    "merged_df.rename(columns={'Unnamed: 0': 'Ortogroup'}, inplace=True)\n",
    "merged_df\n",
    "prot18_ = merged_df[[\"lascID\", \"Ortogroup\", \"geneNameProtein\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Group</th>\n",
       "      <th>geneName</th>\n",
       "      <th>product</th>\n",
       "      <th>lascID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>OG0000264</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_3,Spice_4,Spice_5,Spice_6,Spice_2,Spice_...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC007774</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0  Group                                           geneName  \\\n",
       "262  OG0000264  SpiCE  Spice_3,Spice_4,Spice_5,Spice_6,Spice_2,Spice_...   \n",
       "\n",
       "                  product      lascID  \n",
       "262  hypothetical protein  LASC007774  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_exploded[anno_exploded[\"lascID\"]== \"LASC007774\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## OGs of high loadings also found in silk of gland proteomics data'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:38: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:38: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:38: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\2271155724.py:38: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OG_id</th>\n",
       "      <th>PC1</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>cluster</th>\n",
       "      <th>18protein dataset</th>\n",
       "      <th>silk proteomics</th>\n",
       "      <th>gland proteomics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OG0000282</td>\n",
       "      <td>-0.480160</td>\n",
       "      <td>LASC013610, cd74_2, LASC013607, tg_5, LASC0136...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OG0008243</td>\n",
       "      <td>-0.398308</td>\n",
       "      <td>sp-1339#prd34368.1_4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OG0008607</td>\n",
       "      <td>0.232044</td>\n",
       "      <td>LASC012976</td>\n",
       "      <td>Duct_1, Duct_2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OG0008312</td>\n",
       "      <td>-0.227529</td>\n",
       "      <td>LASC018595</td>\n",
       "      <td>ZoneC_SpiCE-LMa1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OG0000264</td>\n",
       "      <td>0.222516</td>\n",
       "      <td>Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>OG0007089</td>\n",
       "      <td>0.001582</td>\n",
       "      <td>LASC003517</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>OG0006013</td>\n",
       "      <td>0.001425</td>\n",
       "      <td>timm23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>OG0004738</td>\n",
       "      <td>0.000700</td>\n",
       "      <td>LASC004484</td>\n",
       "      <td>Duct_2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>OG0003918</td>\n",
       "      <td>-0.000696</td>\n",
       "      <td>rpl19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>OG0001157</td>\n",
       "      <td>-0.000052</td>\n",
       "      <td>tmem68, tmem68_2</td>\n",
       "      <td>ZoneC_SpiCE-LMa1, nan</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>173 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         OG_id       PC1                                           GeneName  \\\n",
       "0    OG0000282 -0.480160  LASC013610, cd74_2, LASC013607, tg_5, LASC0136...   \n",
       "1    OG0008243 -0.398308                               sp-1339#prd34368.1_4   \n",
       "2    OG0008607  0.232044                                         LASC012976   \n",
       "3    OG0008312 -0.227529                                         LASC018595   \n",
       "4    OG0000264  0.222516  Spice_4, Spice_9, Spice_15, Spice_3, Spice_12,...   \n",
       "..         ...       ...                                                ...   \n",
       "168  OG0007089  0.001582                                         LASC003517   \n",
       "169  OG0006013  0.001425                                             timm23   \n",
       "170  OG0004738  0.000700                                         LASC004484   \n",
       "171  OG0003918 -0.000696                                              rpl19   \n",
       "172  OG0001157 -0.000052                                   tmem68, tmem68_2   \n",
       "\n",
       "                   cluster 18protein dataset silk proteomics gland proteomics  \n",
       "0                      NaN                 -               -                -  \n",
       "1                      NaN                 -               -                -  \n",
       "2           Duct_1, Duct_2                 -               -                -  \n",
       "3         ZoneC_SpiCE-LMa1                 -               -                -  \n",
       "4                      NaN               Yes             Yes                -  \n",
       "..                     ...               ...             ...              ...  \n",
       "168                    NaN                 -               -              Yes  \n",
       "169                    NaN                 -               -                -  \n",
       "170                 Duct_2                 -               -                -  \n",
       "171                    NaN                 -               -              Yes  \n",
       "172  ZoneC_SpiCE-LMa1, nan                 -               -              Yes  \n",
       "\n",
       "[173 rows x 7 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pc = \"PC1\"\n",
    "for df_type in df_list: \n",
    "    for pc in [\"PC1\"]: \n",
    "        \n",
    "        df = df_type.reindex(df_type[pc].abs().sort_values(ascending=False).index)\n",
    "        def check_prot18(row):\n",
    "            values = row['Lascid'].split(',')\n",
    "            if any(value in prot18_['lascID'].values for value in values):\n",
    "                return 'Yes'\n",
    "            else:\n",
    "                return '-'\n",
    "\n",
    "        def check_prot_silk(row):\n",
    "            values = row['Lascid'].split(',')\n",
    "            if any(value in prot_silk['lascID'].values for value in values):\n",
    "                return 'Yes'\n",
    "            else:\n",
    "                return '-'\n",
    "\n",
    "        def check_gland_silk(row):\n",
    "            values = row['GeneName'].split(',')\n",
    "            if any(value in prot_gland['geneName'].values for value in values):\n",
    "                return 'Yes'\n",
    "            else:\n",
    "                return '-'\n",
    "\n",
    "        df = ara_mag_sum.reindex(ara_mag_sum[pc].abs().sort_values(ascending=False).index)\n",
    "\n",
    "\n",
    "        create_full_info_df = df[['OG_id', pc, 'Group', 'GeneName',\n",
    "            'Product', 'Lascid', 'cluster']]\n",
    "\n",
    "        create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
    "        create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
    "        create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
    "\n",
    "create_full_info_df[['OG_id',pc, 'GeneName', 'cluster',\n",
    "        '18protein dataset', 'silk proteomics', 'gland proteomics' ]]\n",
    "        \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{lrlllll}\n",
      "\\toprule\n",
      "OG_id & PC1 & GeneName & geneNameProtein & cluster & 18protein dataset & silk proteomics \\\\\n",
      "\\midrule\n",
      "OG0000282 & -0.480160 & LASC013610, cd74_2, LASC013607, tg_5, LASC013609, LASC013597, LASC013604, LASC013605, LASC013608, nid2, LASC013602, LASC013595 & nan & nan & - & - \\\\\n",
      "OG0008243 & -0.398308 & sp-1339#prd34368.1_4 & nan & nan & - & - \\\\\n",
      "OG0008607 & 0.232044 & LASC012976 & nan & Duct_1, Duct_2 & - & - \\\\\n",
      "OG0008312 & -0.227529 & LASC018595 & nan & ZoneC_SpiCE-LMa1 & - & - \\\\\n",
      "OG0000264 & 0.222516 & Spice_4, Spice_9, Spice_15, Spice_3, Spice_12, Spice_13, Spice_1, Spice_11, Spice_6, Spice_8, Spice_14, Spice_10, Spice_5, Spice_2, Spice_7, Spice_16,Spice_4, Spice_9, Spice_15, Spice_3, Spice_12, Spice_13, Spice_1, Spice_11, Spice_6, Spice_8, Spice_14, Spice_10, Spice_5, Spice_2, Spice_7, Spice_16 & SpiCE-LMa5,nan & nan,nan & Yes & Yes \\\\\n",
      "OG0004717 & -0.195772 & LASC001730, LASC001732, LASC001731, LASC001733 & nan & Duct_1, nan, Duct_2 & - & - \\\\\n",
      "OG0007747 & 0.166421 & cpib & nan & ZoneA_SpiCE-LMa3 & - & - \\\\\n",
      "OG0007860 & 0.161310 & rpl37a_2 & nan & nan & - & - \\\\\n",
      "OG0004409 & 0.158578 & rpl34 & nan & nan & - & - \\\\\n",
      "OG0000189 & -0.136838 & pxd_pxt_3, pxt_2, pxd_5, pxd_pxt_4, pxd_pxt, pxt_4, pxd_pxt_2, pxd_4 & nan & nan & - & Yes \\\\\n",
      "OG0005788 & -0.133239 & rpl27 & nan & nan & - & - \\\\\n",
      "OG0003565 & 0.131620 & LASC016362, LASC019275, LASC006841 & nan & Duct_1, nan, Duct_2 & - & - \\\\\n",
      "OG0000369 & -0.120817 & gld_15, LASC008285, gld_3, gld_19, beta_3 & nan & nan & - & Yes \\\\\n",
      "OG0007102 & -0.103828 & tmem163 & nan & ZoneB_MaSp3, ZoneC_SpiCE-LMa1 & - & - \\\\\n",
      "OG0003330 & -0.100583 & rps24 & nan & nan & - & - \\\\\n",
      "OG0002778 & -0.097265 & Spidroin3 & AmSp-like2 & ZoneB_MaSp3 & Yes & Yes \\\\\n",
      "OG0007367 & -0.093221 & sec61b & nan & nan & - & Yes \\\\\n",
      "OG0001263 & -0.092936 & rplp1, rplp1_2 & nan & nan & - & - \\\\\n",
      "OG0000442 & 0.092170 & hpgd_12, hpgd_3, hpgd_4, hpgd_13, hpgd_9, hpgd_10, hpgd_14, hpgd_15 & nan & nan & - & - \\\\\n",
      "OG0000048 & -0.091822 & LASC007766, LASC015501, LASC011062, LASC022435, LASC012886, LASC011097, LASC013367, LASC015493, LASC010438, LASC012696, LASC021486, LASC017041, LASC006406, LASC011105, LASC012885, LASC005974, LASC010447, LASC020622, LASC013368, LASC005805, LASC011092, LASC022620, LASC013379, LASC015484, LASC001511, LASC015480, LASC015485, LASC013375, LASC011061, LASC021487, LASC010406, LASC010441, LASC012888, LASC013374, LASC011087, LASC015479, LASC011104, LASC015496, LASC005973, LASC010400, LASC012887, LASC010405, LASC010431, LASC015482, LASC010404, LASC011090, LASC015504, LASC010393, LASC012361, LASC013358, LASC011082, LASC015486, LASC010382, LASC011086, LASC017501, LASC011083, LASC022438, LASC015476, LASC022434, LASC015495, LASC010397, LASC015494, LASC022746, LASC011106, LASC010396, LASC015297, LASC013361, LASC011081, LASC015483, LASC011085, LASC011064, LASC013377, LASC011095, LASC022430, LASC007765, LASC010443, LASC015506, LASC010432, LASC017500, LASC015487, LASC022439, LASC013378, LASC010446, LASC010433, LASC005803, LASC011093, LASC022527, LASC013366, LASC011098, LASC003211, LASC022433, LASC013357, LASC022437, LASC015478, LASC013373, LASC005802, LASC015500, LASC007054, LASC011088, LASC008948, LASC018770, LASC013363, LASC017502, LASC010399, LASC008951, LASC010402, LASC010401, LASC011101, LASC015491, LASC010448, LASC011059, LASC011089, LASC013354, LASC017499, LASC011084, LASC011096, LASC007767, LASC010403, LASC012889, LASC008949, LASC013372, LASC010442, LASC015300, LASC011099, LASC015502, LASC022432, LASC013376, LASC011060, LASC013370, LASC010436, LASC011094, LASC007764, LASC011091, LASC010435, LASC022431, LASC011103, LASC010408, LASC015477 & nan & nan, Duct_2 & - & - \\\\\n",
      "\\bottomrule\n",
      "\\end{tabular}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "pc = \"PC1\"\n",
    "for df_type in df_list: \n",
    "    for pc in [\"PC1\"]: \n",
    "        \n",
    "        df = df_type.reindex(df_type[pc].abs().sort_values(ascending=False).index).head(20)\n",
    "\n",
    "        def check_prot18(row):\n",
    "            values = row['Lascid'].split(',')\n",
    "            if any(value in prot18_['lascID'].values for value in values):\n",
    "                return 'Yes'\n",
    "            else:\n",
    "                return '-'\n",
    "\n",
    "        def check_prot_silk(row):\n",
    "            values = row['Lascid'].split(',')\n",
    "            if any(value in prot_silk['lascID'].values for value in values):\n",
    "                return 'Yes'\n",
    "            else:\n",
    "                return '-'\n",
    "\n",
    "        def check_gland_silk(row):\n",
    "            values = row['GeneName'].split(',')\n",
    "            if any(value in prot_gland['geneName'].values for value in values):\n",
    "                return 'Yes'\n",
    "            else:\n",
    "                return '-'\n",
    "\n",
    "        df = ara_mag_sum.reindex(ara_mag_sum[pc].abs().sort_values(ascending=False).index).head(20)\n",
    "\n",
    "\n",
    "        create_full_info_df = df.head(20)[['OG_id', pc, 'Group', 'GeneName',\n",
    "            'Product', 'Lascid', 'cluster']]\n",
    "\n",
    "        create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
    "        create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
    "        create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
    "\n",
    "info_ = create_full_info_df[['OG_id',pc,'GeneName', \"cluster\", \n",
    "        '18protein dataset', 'silk proteomics']]\n",
    "\n",
    "\n",
    "\n",
    "common = pd.merge(info_, silk_og[[\"geneNameProtein\", \"Orthogroup\"]], left_on =\"OG_id\", right_on =\"Orthogroup\" , how = \"left\").iloc[:, :-1]\n",
    "# common[['OG_id', 'PC1', 'GeneName','geneNameProtein', '18protein dataset', 'silk proteomics']]\n",
    "df_grouped = common.groupby('OG_id').agg({\n",
    "    'PC1': \"first\",  \n",
    "    'GeneName': lambda x: ','.join(x.astype(str)), \n",
    "    'geneNameProtein': lambda x: ','.join(x.astype(str)), \n",
    "    'cluster': lambda x: ','.join(x.astype(str)), \n",
    "    '18protein dataset': \"first\", \n",
    "    'silk proteomics': \"first\"\n",
    "}).reset_index()\n",
    "\n",
    "df_grouped = df_grouped.reindex(df_grouped['PC1'].abs().sort_values(ascending=False).index)\n",
    "df_grouped\n",
    "\n",
    "# create_full_info_df[['OG_id',pc,'GeneName', 'cluster',\n",
    "        # '18protein dataset', 'silk proteomics', 'gland proteomics' ]]\n",
    "latex_table = df_grouped.to_latex(index=False)\n",
    "\n",
    "print(latex_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['OG0000050', 'OG0009245', 'OG0013923', 'OG0000171', 'OG0002778',\n",
       "       'OG0007640', 'OG0000494', 'OG0000264', 'OG0006927'], dtype=object)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot18_[\"Ortogroup\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneName</th>\n",
       "      <th>HFIP</th>\n",
       "      <th>LiBr</th>\n",
       "      <th>Urea</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>product</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>lascID</th>\n",
       "      <th>geneName_GOI</th>\n",
       "      <th>Nephila_genome_Class</th>\n",
       "      <th>Group</th>\n",
       "      <th>scGeneName</th>\n",
       "      <th>sc_gene_id</th>\n",
       "      <th>Length</th>\n",
       "      <th>geneNameProtein</th>\n",
       "      <th>silkGroup</th>\n",
       "      <th>Orthogroup</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Spice_8</td>\n",
       "      <td>0.160635</td>\n",
       "      <td>0.014399</td>\n",
       "      <td>0.153889</td>\n",
       "      <td>maker-000078F-est_gff_StringTie-gene-26.2</td>\n",
       "      <td>000078F</td>\n",
       "      <td>6151623</td>\n",
       "      <td>6170329</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC007774</td>\n",
       "      <td>Spice_8</td>\n",
       "      <td>Spice_NMa4</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice-8</td>\n",
       "      <td>maker-000078F-est-gff-StringTie-gene-26.2</td>\n",
       "      <td>147.0</td>\n",
       "      <td>SpiCE-LMa5</td>\n",
       "      <td>Spice-LMa</td>\n",
       "      <td>OG0000264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Spice_15</td>\n",
       "      <td>0.078117</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.059856</td>\n",
       "      <td>maker-000197F-est_gff_StringTie-gene-10.0</td>\n",
       "      <td>000197F</td>\n",
       "      <td>2311869</td>\n",
       "      <td>2326978</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC014000</td>\n",
       "      <td>Spice_15</td>\n",
       "      <td>Spice_NMa4</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice-15</td>\n",
       "      <td>maker-000197F-est-gff-StringTie-gene-10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OG0000264</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    geneName      HFIP      LiBr      Urea  \\\n",
       "23   Spice_8  0.160635  0.014399  0.153889   \n",
       "45  Spice_15  0.078117  0.000000  0.059856   \n",
       "\n",
       "                                      gene_id      Chr    Start      End  \\\n",
       "23  maker-000078F-est_gff_StringTie-gene-26.2  000078F  6151623  6170329   \n",
       "45  maker-000197F-est_gff_StringTie-gene-10.0  000197F  2311869  2326978   \n",
       "\n",
       "                 product uniprot_id      lascID geneName_GOI  \\\n",
       "23  hypothetical protein        NaN  LASC007774      Spice_8   \n",
       "45  hypothetical protein        NaN  LASC014000     Spice_15   \n",
       "\n",
       "   Nephila_genome_Class  Group scGeneName  \\\n",
       "23           Spice_NMa4  SpiCE    Spice-8   \n",
       "45           Spice_NMa4  SpiCE   Spice-15   \n",
       "\n",
       "                                   sc_gene_id  Length geneNameProtein  \\\n",
       "23  maker-000078F-est-gff-StringTie-gene-26.2   147.0      SpiCE-LMa5   \n",
       "45  maker-000197F-est-gff-StringTie-gene-10.0     NaN             NaN   \n",
       "\n",
       "    silkGroup Orthogroup  \n",
       "23  Spice-LMa  OG0000264  \n",
       "45        NaN  OG0000264  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "silk_og[silk_og[[\"silkGroup\", \"Orthogroup\"]][\"Orthogroup\"]== \"OG0000264\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PC1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:39: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:40: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:41: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PC2\n",
      "PC3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:39: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:40: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:41: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:39: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:40: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:41: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:39: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:40: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:41: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:39: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PC4\n",
      "PC5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:40: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_7688\\1755182338.py:41: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PC</th>\n",
       "      <th>18 Protein</th>\n",
       "      <th>Silk Proteomics</th>\n",
       "      <th>Gland Proteomics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PC1</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PC2</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PC3</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PC4</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PC5</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PC  18 Protein  Silk Proteomics  Gland Proteomics\n",
       "0  PC1           6               20                91\n",
       "1  PC2           6               20                91\n",
       "2  PC3           6               20                91\n",
       "3  PC4           6               20                91\n",
       "4  PC5           6               20                91"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "counts = {'PC': [], '18 Protein': [], 'Silk Proteomics': [], 'Gland Proteomics': []}\n",
    "\n",
    "for pc in [\"PC1\", \"PC2\", \"PC3\", \"PC4\", \"PC5\"]:\n",
    "    print(pc)\n",
    "    df = ara_mag_sum.reindex(ara_mag_sum[pc].abs().sort_values(ascending=False).index)\n",
    "    \n",
    "    def check_prot18(row):\n",
    "        values = row['Lascid'].split(',')\n",
    "        if any(value in prot18_['lascID'].values for value in values):\n",
    "            return 'Yes'\n",
    "        else:\n",
    "            return 'No'\n",
    "\n",
    "    def check_prot_silk(row):\n",
    "        values = row['Lascid'].split(',')\n",
    "        if any(value in prot_silk['lascID'].values for value in values):\n",
    "            return 'Yes'\n",
    "        else:\n",
    "            return 'No'\n",
    "\n",
    "    def check_gland_silk(row):\n",
    "        values = row['GeneName'].split(',')\n",
    "        if any(value in prot_gland['geneName'].values for value in values):\n",
    "            return 'Yes'\n",
    "        else:\n",
    "            return 'No'\n",
    "\n",
    "\n",
    "    create_full_info_df = df[['OG_id', 'Lascid', 'GeneName']]\n",
    "    \n",
    "    create_full_info_df['18protein dataset'] = create_full_info_df.apply(check_prot18, axis=1)\n",
    "    create_full_info_df['silk proteomics'] = create_full_info_df.apply(check_prot_silk, axis=1)\n",
    "    create_full_info_df['gland proteomics'] = create_full_info_df.apply(check_gland_silk, axis=1)\n",
    "    \n",
    "\n",
    "    count_yes_18protein = (create_full_info_df[\"18protein dataset\"] == \"Yes\").sum()\n",
    "    count_yes_silk_proteomics = (create_full_info_df[\"silk proteomics\"] == \"Yes\").sum()\n",
    "    count_yes_gland_proteomics = (create_full_info_df[\"gland proteomics\"] == \"Yes\").sum()\n",
    "    \n",
    "    counts['PC'].append(pc)\n",
    "    counts['18 Protein'].append(count_yes_18protein)\n",
    "    counts['Silk Proteomics'].append(count_yes_silk_proteomics)\n",
    "    counts['Gland Proteomics'].append(count_yes_gland_proteomics)\n",
    "counts_df = pd.DataFrame(counts)\n",
    "\n",
    "counts_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OG_id</th>\n",
       "      <th>Lascid</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>18protein dataset</th>\n",
       "      <th>silk proteomics</th>\n",
       "      <th>gland proteomics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OG0008607</td>\n",
       "      <td>LASC012976</td>\n",
       "      <td>LASC012976</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>OG0003565</td>\n",
       "      <td>LASC006841,LASC016362,LASC019275</td>\n",
       "      <td>LASC016362, LASC019275, LASC006841</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>OG0005938</td>\n",
       "      <td>LASC000884</td>\n",
       "      <td>rpl21</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>OG0007367</td>\n",
       "      <td>LASC021718</td>\n",
       "      <td>sec61b</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>OG0000048</td>\n",
       "      <td>LASC001511,LASC003211,LASC005802,LASC005803,LA...</td>\n",
       "      <td>LASC007766, LASC015501, LASC011062, LASC022435...</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>OG0001429</td>\n",
       "      <td>LASC010675,LASC021902</td>\n",
       "      <td>rpl9_2, rpl9</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>OG0004676</td>\n",
       "      <td>LASC019543</td>\n",
       "      <td>tsnax</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>OG0003221</td>\n",
       "      <td>LASC011010</td>\n",
       "      <td>phgdh</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>OG0006712</td>\n",
       "      <td>LASC003348</td>\n",
       "      <td>kdelr2</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>OG0001207</td>\n",
       "      <td>LASC009993,LASC009996</td>\n",
       "      <td>rps2_2, rps2</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>173 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         OG_id                                             Lascid  \\\n",
       "2    OG0008607                                         LASC012976   \n",
       "11   OG0003565                   LASC006841,LASC016362,LASC019275   \n",
       "36   OG0005938                                         LASC000884   \n",
       "16   OG0007367                                         LASC021718   \n",
       "19   OG0000048  LASC001511,LASC003211,LASC005802,LASC005803,LA...   \n",
       "..         ...                                                ...   \n",
       "75   OG0001429                              LASC010675,LASC021902   \n",
       "143  OG0004676                                         LASC019543   \n",
       "105  OG0003221                                         LASC011010   \n",
       "103  OG0006712                                         LASC003348   \n",
       "59   OG0001207                              LASC009993,LASC009996   \n",
       "\n",
       "                                              GeneName 18protein dataset  \\\n",
       "2                                           LASC012976                No   \n",
       "11                  LASC016362, LASC019275, LASC006841                No   \n",
       "36                                               rpl21                No   \n",
       "16                                              sec61b                No   \n",
       "19   LASC007766, LASC015501, LASC011062, LASC022435...                No   \n",
       "..                                                 ...               ...   \n",
       "75                                        rpl9_2, rpl9                No   \n",
       "143                                              tsnax                No   \n",
       "105                                              phgdh                No   \n",
       "103                                             kdelr2                No   \n",
       "59                                        rps2_2, rps2                No   \n",
       "\n",
       "    silk proteomics gland proteomics  \n",
       "2                No               No  \n",
       "11               No               No  \n",
       "36               No              Yes  \n",
       "16              Yes              Yes  \n",
       "19               No               No  \n",
       "..              ...              ...  \n",
       "75               No              Yes  \n",
       "143              No               No  \n",
       "105              No              Yes  \n",
       "103              No               No  \n",
       "59               No              Yes  \n",
       "\n",
       "[173 rows x 6 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_full_info_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'i' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[33], line 9\u001b[0m\n\u001b[0;32m      7\u001b[0m \u001b[38;5;66;03m# joined_full_gland = pd.merge(prot_gland, df_20_exploded, left_on='geneName', right_on='GeneName', how='inner')\u001b[39;00m\n\u001b[0;32m      8\u001b[0m joined_gland \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mmerge(prot_gland, df_20_exploded, left_on\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgeneName\u001b[39m\u001b[38;5;124m'\u001b[39m, right_on\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGeneName\u001b[39m\u001b[38;5;124m'\u001b[39m, how\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124minner\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m----> 9\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdf_names[\u001b[43mi\u001b[49m]\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m, to gland proteomics\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m     10\u001b[0m \u001b[38;5;28mprint\u001b[39m(joined_gland[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mOG_id\u001b[39m\u001b[38;5;124m\"\u001b[39m]\u001b[38;5;241m.\u001b[39munique())\n\u001b[0;32m     12\u001b[0m df_20 \u001b[38;5;241m=\u001b[39m df\u001b[38;5;241m.\u001b[39mhead(\u001b[38;5;241m20\u001b[39m)\n",
      "\u001b[1;31mNameError\u001b[0m: name 'i' is not defined"
     ]
    }
   ],
   "source": [
    "df = ara_mag_sum.reindex(ara_mag_sum['PC1'].abs().sort_values(ascending=False).index).head(20)\n",
    "df_20 =df\n",
    "\n",
    "df_20['GeneName'] = df_20['GeneName'].str.split(',')\n",
    "df_20_exploded = df_20.explode('GeneName')\n",
    "\n",
    "# joined_full_gland = pd.merge(prot_gland, df_20_exploded, left_on='geneName', right_on='GeneName', how='inner')\n",
    "joined_gland = pd.merge(prot_gland, df_20_exploded, left_on='geneName', right_on='GeneName', how='inner')\n",
    "print(f\"{df_names[i]}, to gland proteomics\")\n",
    "print(joined_gland[\"OG_id\"].unique())\n",
    "\n",
    "df_20 = df.head(20)\n",
    "\n",
    "df_20['Lascid'] = df_20['Lascid'].str.split(',')\n",
    "df_20_exploded = df_20.explode('Lascid')\n",
    "\n",
    "joined_silk = pd.merge(prot_silk, df_20_exploded, left_on='lascID', right_on='Lascid', how='inner')\n",
    "print(f\"{df_names[i]}, to silk proteomics\")\n",
    "print(joined_silk[\"OG_id\"].unique())\n",
    "joined_silk[['OG_id', \"Lascid\", \"cluster\", \"Group_y\", \"geneName\", \"scGeneName\", \"geneName_GOI\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_6580\\19846268.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_20['GeneName'] = df_20['GeneName'].str.split(',')\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_6580\\19846268.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_20['Lascid'] = df_20['Lascid'].str.split(',')\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_6580\\19846268.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_20['GeneName'] = df_20['GeneName'].str.split(',')\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_6580\\19846268.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_20['Lascid'] = df_20['Lascid'].str.split(',')\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_6580\\19846268.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_20['GeneName'] = df_20['GeneName'].str.split(',')\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_6580\\19846268.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_20['Lascid'] = df_20['Lascid'].str.split(',')\n",
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_6580\\19846268.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_20['GeneName'] = df_20['GeneName'].str.split(',')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ara_mag_sum, to gland proteomics\n",
      "['OG0002778' 'OG0007747' 'OG0003330' 'OG0005788' 'OG0007367' 'OG0007860'\n",
      " 'OG0004409']\n",
      "ara_mag_sum, to silk proteomics\n",
      "['OG0002778' 'OG0000369' 'OG0000264' 'OG0000189' 'OG0007367']\n",
      "mag_ara_bool_1, to gland proteomics\n",
      "['OG0004629' 'OG0002778' 'OG0008612' 'OG0007640' 'OG0009320' 'OG0002604'\n",
      " 'OG0008177' 'OG0008600' 'OG0005938' 'OG0004409' 'OG0002039']\n",
      "mag_ara_bool_1, to silk proteomics\n",
      "['OG0002778' 'OG0007640' 'OG0002604']\n",
      "mag_ara_bool_2, to gland proteomics\n",
      "['OG0004629' 'OG0002778' 'OG0003330' 'OG0008612' 'OG0009320' 'OG0004318'\n",
      " 'OG0002604' 'OG0007860' 'OG0008600' 'OG0005938' 'OG0002319' 'OG0002039']\n",
      "mag_ara_bool_2, to silk proteomics\n",
      "['OG0002778' 'OG0006927' 'OG0004318' 'OG0002604']\n",
      "ara_mag_GC, to gland proteomics\n",
      "['OG0000050' 'OG0000055' 'OG0000369' 'OG0000609' 'OG0000073' 'OG0000022'\n",
      " 'OG0000081' 'OG0000127']\n",
      "ara_mag_GC, to silk proteomics\n",
      "['OG0000050' 'OG0000369' 'OG0000264' 'OG0000078']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\46705\\AppData\\Local\\Temp\\ipykernel_6580\\19846268.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_20['Lascid'] = df_20['Lascid'].str.split(',')\n"
     ]
    }
   ],
   "source": [
    "for i, df in enumerate(df_list): \n",
    "    df_20 = df.head(20)\n",
    "\n",
    "    df_20['GeneName'] = df_20['GeneName'].str.split(',')\n",
    "    df_20_exploded = df_20.explode('GeneName')\n",
    "    \n",
    "    # joined_full_gland = pd.merge(prot_gland, df_20_exploded, left_on='geneName', right_on='GeneName', how='inner')\n",
    "    joined_gland = pd.merge(prot_gland, df_20_exploded, left_on='geneName', right_on='GeneName', how='inner')\n",
    "    print(f\"{df_names[i]}, to gland proteomics\")\n",
    "    print(joined_gland[\"OG_id\"].unique())\n",
    "\n",
    "    df_20 = df.head(20)\n",
    "\n",
    "    df_20['Lascid'] = df_20['Lascid'].str.split(',')\n",
    "    df_20_exploded = df_20.explode('Lascid')\n",
    "\n",
    "    joined_silk = pd.merge(prot_silk, df_20_exploded, left_on='lascID', right_on='Lascid', how='inner')\n",
    "    print(f\"{df_names[i]}, to silk proteomics\")\n",
    "    print(joined_silk[\"OG_id\"].unique())\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneName</th>\n",
       "      <th>HFIP</th>\n",
       "      <th>LiBr</th>\n",
       "      <th>Urea</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>product</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>...</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>Group_y</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Product</th>\n",
       "      <th>Lascid</th>\n",
       "      <th>cluster</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MaSp1a</td>\n",
       "      <td>22.432921</td>\n",
       "      <td>17.814666</td>\n",
       "      <td>23.677962</td>\n",
       "      <td>evm-000033F-processed-gene-27.15-GeneA</td>\n",
       "      <td>000033F</td>\n",
       "      <td>6104352</td>\n",
       "      <td>6121610</td>\n",
       "      <td>Major, ampullate, spidroin, 1, variant, 1, (Fr...</td>\n",
       "      <td>A0A2S2B4B0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004271</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MaSp3a</td>\n",
       "      <td>13.046241</td>\n",
       "      <td>14.304353</td>\n",
       "      <td>16.506424</td>\n",
       "      <td>evm-000033F-processed-gene-27.15-GeneD</td>\n",
       "      <td>000033F</td>\n",
       "      <td>6474253</td>\n",
       "      <td>6482733</td>\n",
       "      <td>Major, ampullate, spidroin, 1, variant, 1, (Fr...</td>\n",
       "      <td>A0A2S2B4B0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004275</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MaSp1b</td>\n",
       "      <td>11.515409</td>\n",
       "      <td>7.703781</td>\n",
       "      <td>12.856568</td>\n",
       "      <td>evm-000033F-processed-gene-27.15-GeneB</td>\n",
       "      <td>000033F</td>\n",
       "      <td>6189272</td>\n",
       "      <td>6198799</td>\n",
       "      <td>Major, ampullate, spidroin, 1, variant, 1, (Fr...</td>\n",
       "      <td>A0A2S2B4B0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004273</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MaSp3b</td>\n",
       "      <td>7.154938</td>\n",
       "      <td>8.087669</td>\n",
       "      <td>8.801191</td>\n",
       "      <td>evm-000033F-processed-gene-27.15-GeneE</td>\n",
       "      <td>000033F</td>\n",
       "      <td>6540284</td>\n",
       "      <td>6549234</td>\n",
       "      <td>Major, ampullate, spidroin, 1, variant, 1, (Fr...</td>\n",
       "      <td>A0A2S2B4B0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004276</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MaSp2c</td>\n",
       "      <td>5.650715</td>\n",
       "      <td>5.309158</td>\n",
       "      <td>5.080537</td>\n",
       "      <td>evm-000033F-processed-gene-24.21-GeneA</td>\n",
       "      <td>000033F</td>\n",
       "      <td>5776789</td>\n",
       "      <td>5819345</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein</td>\n",
       "      <td>PRD27696.1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004269</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MaSp2f</td>\n",
       "      <td>4.156333</td>\n",
       "      <td>3.800549</td>\n",
       "      <td>4.504349</td>\n",
       "      <td>maker-000175F-exonerate_protein2genome-gene-8....</td>\n",
       "      <td>000175F</td>\n",
       "      <td>1857400</td>\n",
       "      <td>1874685</td>\n",
       "      <td>Major, ampullate, spidroin, 2, variant, 2, (Fr...</td>\n",
       "      <td>A0A2S2B4G1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC012922</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MaSp1c</td>\n",
       "      <td>3.449324</td>\n",
       "      <td>1.709131</td>\n",
       "      <td>4.009279</td>\n",
       "      <td>evm-000033F-processed-gene-27.15-GeneC</td>\n",
       "      <td>000033F</td>\n",
       "      <td>6258576</td>\n",
       "      <td>6268628</td>\n",
       "      <td>Major, ampullate, spidroin, 1, variant, 1, (Fr...</td>\n",
       "      <td>A0A2S2B4B0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004274</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MaSp2b</td>\n",
       "      <td>3.636470</td>\n",
       "      <td>3.133324</td>\n",
       "      <td>3.725936</td>\n",
       "      <td>evm-000033F-processed-gene-24.21-GeneB</td>\n",
       "      <td>000033F</td>\n",
       "      <td>5626769</td>\n",
       "      <td>5627335</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein</td>\n",
       "      <td>PRD27696.1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004267</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MaSp2e</td>\n",
       "      <td>2.859585</td>\n",
       "      <td>1.910398</td>\n",
       "      <td>3.031373</td>\n",
       "      <td>maker-000175F-exonerate_protein2genome-gene-8....</td>\n",
       "      <td>000175F</td>\n",
       "      <td>1701320</td>\n",
       "      <td>1716556</td>\n",
       "      <td>Major, ampullate, spidroin, 2, variant, 2, (Fr...</td>\n",
       "      <td>A0A2S2B4G1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC012920</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MaSp4</td>\n",
       "      <td>0.895281</td>\n",
       "      <td>1.332501</td>\n",
       "      <td>1.085557</td>\n",
       "      <td>evm-000033F-processed-gene-22.11</td>\n",
       "      <td>000033F</td>\n",
       "      <td>5379690</td>\n",
       "      <td>5392488</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein</td>\n",
       "      <td>PRD27696.1</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>beta_3</td>\n",
       "      <td>0.242286</td>\n",
       "      <td>0.235661</td>\n",
       "      <td>0.395209</td>\n",
       "      <td>evm-000086F-processed-gene-4.0</td>\n",
       "      <td>000086F</td>\n",
       "      <td>1024637</td>\n",
       "      <td>1086127</td>\n",
       "      <td>Oxygen-dependent choline dehydrogenase</td>\n",
       "      <td>Q985M5</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.064922</td>\n",
       "      <td>0.176620</td>\n",
       "      <td>-0.111129</td>\n",
       "      <td>-0.081020</td>\n",
       "      <td>0.060826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>beta_3, gld_3, gld_19, gld_15, LASC008285</td>\n",
       "      <td>Glucose dehydrogenase [FAD%2C quinone],hypothe...</td>\n",
       "      <td>LASC008287</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Spice_8</td>\n",
       "      <td>0.160635</td>\n",
       "      <td>0.014399</td>\n",
       "      <td>0.153889</td>\n",
       "      <td>maker-000078F-est_gff_StringTie-gene-26.2</td>\n",
       "      <td>000078F</td>\n",
       "      <td>6151623</td>\n",
       "      <td>6170329</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039464</td>\n",
       "      <td>0.053178</td>\n",
       "      <td>0.082237</td>\n",
       "      <td>0.480030</td>\n",
       "      <td>-0.195722</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_14, Spice_10, Spice_7, Spice_8, Spice_5,...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC007774</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>LASC002536</td>\n",
       "      <td>0.121379</td>\n",
       "      <td>0.095573</td>\n",
       "      <td>0.118633</td>\n",
       "      <td>maker-000015F-augustus-gene-50.0</td>\n",
       "      <td>000015F</td>\n",
       "      <td>11717636</td>\n",
       "      <td>11752337</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.064206</td>\n",
       "      <td>0.225648</td>\n",
       "      <td>0.085521</td>\n",
       "      <td>-0.147190</td>\n",
       "      <td>0.210642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC002408, ache, LASC000874, LASC017753, LASC...</td>\n",
       "      <td>hypothetical protein,Programmed cell death pro...</td>\n",
       "      <td>LASC002536</td>\n",
       "      <td>nan, ZoneABC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Spice_15</td>\n",
       "      <td>0.078117</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.059856</td>\n",
       "      <td>maker-000197F-est_gff_StringTie-gene-10.0</td>\n",
       "      <td>000197F</td>\n",
       "      <td>2311869</td>\n",
       "      <td>2326978</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.039464</td>\n",
       "      <td>0.053178</td>\n",
       "      <td>0.082237</td>\n",
       "      <td>0.480030</td>\n",
       "      <td>-0.195722</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_14, Spice_10, Spice_7, Spice_8, Spice_5,...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC014000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MiSpb</td>\n",
       "      <td>1.979309</td>\n",
       "      <td>4.533598</td>\n",
       "      <td>0.046363</td>\n",
       "      <td>evm-000071F-processed-gene-30.32-GeneB</td>\n",
       "      <td>000071F</td>\n",
       "      <td>7038336</td>\n",
       "      <td>7051176</td>\n",
       "      <td>hypothetical, protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC007257</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>LASC000874</td>\n",
       "      <td>0.026492</td>\n",
       "      <td>0.014399</td>\n",
       "      <td>0.032266</td>\n",
       "      <td>maker-000003F-est_gff_StringTie-gene-77.3</td>\n",
       "      <td>000003F</td>\n",
       "      <td>18364566</td>\n",
       "      <td>18417425</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.064206</td>\n",
       "      <td>0.225648</td>\n",
       "      <td>0.085521</td>\n",
       "      <td>-0.147190</td>\n",
       "      <td>0.210642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC002408, ache, LASC000874, LASC017753, LASC...</td>\n",
       "      <td>hypothetical protein,Programmed cell death pro...</td>\n",
       "      <td>LASC000874</td>\n",
       "      <td>nan, ZoneABC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      geneName       HFIP       LiBr       Urea  \\\n",
       "0       MaSp1a  22.432921  17.814666  23.677962   \n",
       "1       MaSp3a  13.046241  14.304353  16.506424   \n",
       "2       MaSp1b  11.515409   7.703781  12.856568   \n",
       "3       MaSp3b   7.154938   8.087669   8.801191   \n",
       "4       MaSp2c   5.650715   5.309158   5.080537   \n",
       "5       MaSp2f   4.156333   3.800549   4.504349   \n",
       "6       MaSp1c   3.449324   1.709131   4.009279   \n",
       "7       MaSp2b   3.636470   3.133324   3.725936   \n",
       "8       MaSp2e   2.859585   1.910398   3.031373   \n",
       "9        MaSp4   0.895281   1.332501   1.085557   \n",
       "10      beta_3   0.242286   0.235661   0.395209   \n",
       "11     Spice_8   0.160635   0.014399   0.153889   \n",
       "12  LASC002536   0.121379   0.095573   0.118633   \n",
       "13    Spice_15   0.078117   0.000000   0.059856   \n",
       "14       MiSpb   1.979309   4.533598   0.046363   \n",
       "15  LASC000874   0.026492   0.014399   0.032266   \n",
       "\n",
       "                                              gene_id      Chr     Start  \\\n",
       "0              evm-000033F-processed-gene-27.15-GeneA  000033F   6104352   \n",
       "1              evm-000033F-processed-gene-27.15-GeneD  000033F   6474253   \n",
       "2              evm-000033F-processed-gene-27.15-GeneB  000033F   6189272   \n",
       "3              evm-000033F-processed-gene-27.15-GeneE  000033F   6540284   \n",
       "4              evm-000033F-processed-gene-24.21-GeneA  000033F   5776789   \n",
       "5   maker-000175F-exonerate_protein2genome-gene-8....  000175F   1857400   \n",
       "6              evm-000033F-processed-gene-27.15-GeneC  000033F   6258576   \n",
       "7              evm-000033F-processed-gene-24.21-GeneB  000033F   5626769   \n",
       "8   maker-000175F-exonerate_protein2genome-gene-8....  000175F   1701320   \n",
       "9                    evm-000033F-processed-gene-22.11  000033F   5379690   \n",
       "10                     evm-000086F-processed-gene-4.0  000086F   1024637   \n",
       "11          maker-000078F-est_gff_StringTie-gene-26.2  000078F   6151623   \n",
       "12                   maker-000015F-augustus-gene-50.0  000015F  11717636   \n",
       "13          maker-000197F-est_gff_StringTie-gene-10.0  000197F   2311869   \n",
       "14             evm-000071F-processed-gene-30.32-GeneB  000071F   7038336   \n",
       "15          maker-000003F-est_gff_StringTie-gene-77.3  000003F  18364566   \n",
       "\n",
       "         End                                            product  uniprot_id  \\\n",
       "0    6121610  Major, ampullate, spidroin, 1, variant, 1, (Fr...  A0A2S2B4B0   \n",
       "1    6482733  Major, ampullate, spidroin, 1, variant, 1, (Fr...  A0A2S2B4B0   \n",
       "2    6198799  Major, ampullate, spidroin, 1, variant, 1, (Fr...  A0A2S2B4B0   \n",
       "3    6549234  Major, ampullate, spidroin, 1, variant, 1, (Fr...  A0A2S2B4B0   \n",
       "4    5819345                      MaSp-f(1)#PRD27696.1, protein  PRD27696.1   \n",
       "5    1874685  Major, ampullate, spidroin, 2, variant, 2, (Fr...  A0A2S2B4G1   \n",
       "6    6268628  Major, ampullate, spidroin, 1, variant, 1, (Fr...  A0A2S2B4B0   \n",
       "7    5627335                      MaSp-f(1)#PRD27696.1, protein  PRD27696.1   \n",
       "8    1716556  Major, ampullate, spidroin, 2, variant, 2, (Fr...  A0A2S2B4G1   \n",
       "9    5392488                      MaSp-f(1)#PRD27696.1, protein  PRD27696.1   \n",
       "10   1086127             Oxygen-dependent choline dehydrogenase      Q985M5   \n",
       "11   6170329                               hypothetical protein         NaN   \n",
       "12  11752337                               hypothetical protein         NaN   \n",
       "13   2326978                               hypothetical protein         NaN   \n",
       "14   7051176                              hypothetical, protein         NaN   \n",
       "15  18417425                               hypothetical protein         NaN   \n",
       "\n",
       "    ...       PC1       PC2       PC3       PC4       PC5   Group_y  \\\n",
       "0   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "1   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "2   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "3   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "4   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "5   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "6   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "7   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "8   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "9   ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "10  ... -0.064922  0.176620 -0.111129 -0.081020  0.060826       NaN   \n",
       "11  ... -0.039464  0.053178  0.082237  0.480030 -0.195722     SpiCE   \n",
       "12  ... -0.064206  0.225648  0.085521 -0.147190  0.210642       NaN   \n",
       "13  ... -0.039464  0.053178  0.082237  0.480030 -0.195722     SpiCE   \n",
       "14  ... -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "15  ... -0.064206  0.225648  0.085521 -0.147190  0.210642       NaN   \n",
       "\n",
       "                                             GeneName  \\\n",
       "0   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "1   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "2   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "3   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "4   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "5   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "6   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "7   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "8   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "9   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "10          beta_3, gld_3, gld_19, gld_15, LASC008285   \n",
       "11  Spice_14, Spice_10, Spice_7, Spice_8, Spice_5,...   \n",
       "12  LASC002408, ache, LASC000874, LASC017753, LASC...   \n",
       "13  Spice_14, Spice_10, Spice_7, Spice_8, Spice_5,...   \n",
       "14  MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "15  LASC002408, ache, LASC000874, LASC017753, LASC...   \n",
       "\n",
       "                                              Product      Lascid  \\\n",
       "0   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC004271   \n",
       "1   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC004275   \n",
       "2   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC004273   \n",
       "3   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC004276   \n",
       "4   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC004269   \n",
       "5   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC012922   \n",
       "6   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC004274   \n",
       "7   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC004267   \n",
       "8   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC012920   \n",
       "9   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC004263   \n",
       "10  Glucose dehydrogenase [FAD%2C quinone],hypothe...  LASC008287   \n",
       "11                               hypothetical protein  LASC007774   \n",
       "12  hypothetical protein,Programmed cell death pro...  LASC002536   \n",
       "13                               hypothetical protein  LASC014000   \n",
       "14  MaSp-f(1)#PRD27696.1, protein,Major, ampullate...  LASC007257   \n",
       "15  hypothetical protein,Programmed cell death pro...  LASC000874   \n",
       "\n",
       "                                              cluster  \n",
       "0   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "1   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "2   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "3   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "4   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "5   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "6   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "7   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "8   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "9   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "10                                                NaN  \n",
       "11                                                NaN  \n",
       "12                                       nan, ZoneABC  \n",
       "13                                                NaN  \n",
       "14  ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  \n",
       "15                                       nan, ZoneABC  \n",
       "\n",
       "[16 rows x 30 columns]"
      ]
     },
     "execution_count": 263,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "joined_silk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>geneName</th>\n",
       "      <th>percent</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>geneNameProtein</th>\n",
       "      <th>silkGroup</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [geneName, percent, gene_id, geneNameProtein, silkGroup]\n",
       "Index: []"
      ]
     },
     "execution_count": 267,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_gland[prot_gland['geneName'].isin([\"gld_15\", \"pxd_2\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_df.to_csv(\"C:/Users/46705/Documents/SpiderSilk/data/proteins_18.csv\", index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['OG0000050', 'OG0009245', 'OG0013923', 'OG0000171', 'OG0002778',\n",
       "       'OG0007640', 'OG0000494', 'OG0000264', 'OG0006927'], dtype=object)"
      ]
     },
     "execution_count": 274,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot18_[\"Ortogroup\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lascID</th>\n",
       "      <th>Ortogroup</th>\n",
       "      <th>geneNameProtein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LASC004271</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp1a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LASC004275</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp3a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LASC004273</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp1b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LASC004269</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp2c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LASC004274</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp1c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>LASC004276</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp3b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>LASC004267</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp2b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>LASC012922</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp2f</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>LASC012920</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp2e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>LASC013229</td>\n",
       "      <td>OG0009245</td>\n",
       "      <td>SpiCE-LMa1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>LASC004263</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>LASC012835</td>\n",
       "      <td>OG0013923</td>\n",
       "      <td>SpiCE-LMa2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>LASC000319</td>\n",
       "      <td>OG0000171</td>\n",
       "      <td>SpiCE-LMa3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>LASC020272</td>\n",
       "      <td>OG0002778</td>\n",
       "      <td>AmSp-like2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>LASC000723</td>\n",
       "      <td>OG0007640</td>\n",
       "      <td>AmSp-like1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>LASC017765</td>\n",
       "      <td>OG0000494</td>\n",
       "      <td>SpiCE-LMa4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>LASC007774</td>\n",
       "      <td>OG0000264</td>\n",
       "      <td>SpiCE-LMa5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>LASC009843</td>\n",
       "      <td>OG0006927</td>\n",
       "      <td>SpiCE-LMa6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        lascID  Ortogroup geneNameProtein\n",
       "0   LASC004271  OG0000050          MaSp1a\n",
       "1   LASC004275  OG0000050          MaSp3a\n",
       "2   LASC004273  OG0000050          MaSp1b\n",
       "3   LASC004269  OG0000050          MaSp2c\n",
       "4   LASC004274  OG0000050          MaSp1c\n",
       "5   LASC004276  OG0000050          MaSp3b\n",
       "6   LASC004267  OG0000050          MaSp2b\n",
       "7   LASC012922  OG0000050          MaSp2f\n",
       "8   LASC012920  OG0000050          MaSp2e\n",
       "9   LASC013229  OG0009245      SpiCE-LMa1\n",
       "10  LASC004263  OG0000050           MaSp4\n",
       "11  LASC012835  OG0013923      SpiCE-LMa2\n",
       "12  LASC000319  OG0000171      SpiCE-LMa3\n",
       "13  LASC020272  OG0002778      AmSp-like2\n",
       "14  LASC000723  OG0007640      AmSp-like1\n",
       "15  LASC017765  OG0000494      SpiCE-LMa4\n",
       "16  LASC007774  OG0000264      SpiCE-LMa5\n",
       "17  LASC009843  OG0006927      SpiCE-LMa6"
      ]
     },
     "execution_count": 275,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot18_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OG_id</th>\n",
       "      <th>PC1</th>\n",
       "      <th>PC2</th>\n",
       "      <th>PC3</th>\n",
       "      <th>PC4</th>\n",
       "      <th>PC5</th>\n",
       "      <th>Group</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>Product</th>\n",
       "      <th>Lascid</th>\n",
       "      <th>cluster</th>\n",
       "      <th>lascID</th>\n",
       "      <th>Ortogroup</th>\n",
       "      <th>geneNameProtein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC004271</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp1a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC004275</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp3a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC004273</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp1b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC004269</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp2c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC004274</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp1c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC004276</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp3b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC004267</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp2b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC012922</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp2f</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC012920</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp2e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>OG0000050</td>\n",
       "      <td>-0.031896</td>\n",
       "      <td>0.120426</td>\n",
       "      <td>0.195810</td>\n",
       "      <td>-0.075735</td>\n",
       "      <td>0.154411</td>\n",
       "      <td>Spidroin</td>\n",
       "      <td>MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...</td>\n",
       "      <td>MaSp-f(1)#PRD27696.1, protein,Major, ampullate...</td>\n",
       "      <td>LASC004263,LASC004265,LASC004267,LASC004269,LA...</td>\n",
       "      <td>ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...</td>\n",
       "      <td>LASC004263</td>\n",
       "      <td>OG0000050</td>\n",
       "      <td>MaSp4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>OG0000264</td>\n",
       "      <td>-0.039464</td>\n",
       "      <td>0.053178</td>\n",
       "      <td>0.082237</td>\n",
       "      <td>0.480030</td>\n",
       "      <td>-0.195722</td>\n",
       "      <td>SpiCE</td>\n",
       "      <td>Spice_14, Spice_10, Spice_7, Spice_8, Spice_5,...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>LASC001444,LASC001445,LASC001446,LASC001447,LA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC007774</td>\n",
       "      <td>OG0000264</td>\n",
       "      <td>SpiCE-LMa5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        OG_id       PC1       PC2       PC3       PC4       PC5     Group  \\\n",
       "0   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "1   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "2   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "3   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "4   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "5   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "6   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "7   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "8   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "9   OG0000050 -0.031896  0.120426  0.195810 -0.075735  0.154411  Spidroin   \n",
       "10  OG0000264 -0.039464  0.053178  0.082237  0.480030 -0.195722     SpiCE   \n",
       "\n",
       "                                             GeneName  \\\n",
       "0   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "1   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "2   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "3   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "4   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "5   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "6   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "7   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "8   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "9   MaSp3b, MaSp2f, MaSp2d, MaSp3a, MaSp2c, MiSpb,...   \n",
       "10  Spice_14, Spice_10, Spice_7, Spice_8, Spice_5,...   \n",
       "\n",
       "                                              Product  \\\n",
       "0   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "1   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "2   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "3   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "4   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "5   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "6   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "7   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "8   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "9   MaSp-f(1)#PRD27696.1, protein,Major, ampullate...   \n",
       "10                               hypothetical protein   \n",
       "\n",
       "                                               Lascid  \\\n",
       "0   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "1   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "2   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "3   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "4   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "5   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "6   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "7   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "8   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "9   LASC004263,LASC004265,LASC004267,LASC004269,LA...   \n",
       "10  LASC001444,LASC001445,LASC001446,LASC001447,LA...   \n",
       "\n",
       "                                              cluster      lascID  Ortogroup  \\\n",
       "0   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC004271  OG0000050   \n",
       "1   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC004275  OG0000050   \n",
       "2   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC004273  OG0000050   \n",
       "3   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC004269  OG0000050   \n",
       "4   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC004274  OG0000050   \n",
       "5   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC004276  OG0000050   \n",
       "6   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC004267  OG0000050   \n",
       "7   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC012922  OG0000050   \n",
       "8   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC012920  OG0000050   \n",
       "9   ZoneB_MaSp3, nan, ZoneABC, ZoneA_MaSp1, ZoneA_...  LASC004263  OG0000050   \n",
       "10                                                NaN  LASC007774  OG0000264   \n",
       "\n",
       "   geneNameProtein  \n",
       "0           MaSp1a  \n",
       "1           MaSp3a  \n",
       "2           MaSp1b  \n",
       "3           MaSp2c  \n",
       "4           MaSp1c  \n",
       "5           MaSp3b  \n",
       "6           MaSp2b  \n",
       "7           MaSp2f  \n",
       "8           MaSp2e  \n",
       "9            MaSp4  \n",
       "10      SpiCE-LMa5  "
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "silk_top_20 = []\n",
    "for i, df in enumerate(df_list): \n",
    "    df_20 = df.head(20)\n",
    "    merged_df = pd.merge(df_20, prot18_, how='inner', left_on='OG_id', right_on='Ortogroup')\n",
    "    silk_top_20.append(merged_df)\n",
    "\n",
    "df_list = [ara_mag_sum, mag_ara_bool_1, mag_ara_bool_2, ara_mag_GC]\n",
    "silk_top_20[3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import pandas as pd\n",
    "# from scipy.stats import f_oneway\n",
    "\n",
    "# # Example DataFrames\n",
    "\n",
    "# # Calculate fractions\n",
    "# fraction_df = ara_mag_sum_exploded[ara_mag_sum_exploded['Lascid'].isin(prot_silk['lascID'])].shape[0] / ara_mag_sum_exploded.shape[0]\n",
    "# fraction_df_part = ara_mag_sum_20_exploded[ara_mag_sum_20_exploded['Lascid'].isin(prot_silk['lascID'])].shape[0] / ara_mag_sum_20_exploded.shape[0]\n",
    "\n",
    "# # Perform ANOVA test\n",
    "# f_value, p_value = f_oneway([fraction_df], [fraction_df_part])\n",
    "# print(p_value)\n",
    "\n",
    "# # Interpret the result\n",
    "# alpha = 0.05\n",
    "# if p_value < alpha:\n",
    "#     print(\"There is a significant difference in the fraction of values between the DataFrames.\")\n",
    "# else:\n",
    "#     print(\"There is no significant difference in the fraction of values between the DataFrames.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Orthogroup</th>\n",
       "      <th>ID</th>\n",
       "      <th>idv_id</th>\n",
       "      <th>DRR</th>\n",
       "      <th>species_x</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Family</th>\n",
       "      <th>OG0000013</th>\n",
       "      <th>OG0000022</th>\n",
       "      <th>OG0000031</th>\n",
       "      <th>...</th>\n",
       "      <th>five_percent_weightloss</th>\n",
       "      <th>ten_percent_weightloss</th>\n",
       "      <th>crystallinity</th>\n",
       "      <th>birefringence</th>\n",
       "      <th>birefringence_sd</th>\n",
       "      <th>diameter</th>\n",
       "      <th>diameter_sd</th>\n",
       "      <th>water_content</th>\n",
       "      <th>supercontraction</th>\n",
       "      <th>supercontraction_sd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Group</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>geneName</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cyp4c3_3,cyp4v2_cyp4c3_5,cyp4v2_23,cyp4v2_3,cy...</td>\n",
       "      <td>LASC002598,LASC010177,LASC010178,LASC010181,LA...</td>\n",
       "      <td>jhamt_16,LASC005293,jhamt_8,jhamt_11,jhamt_5,j...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>product</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cytochrome P450 4c3,Cytochrome P450 4V2,Cytoch...</td>\n",
       "      <td>hypothetical protein</td>\n",
       "      <td>Juvenile hormone acid O-methyltransferase,hypo...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>lascID</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LASC003001,LASC003994,LASC003995,LASC005732,LA...</td>\n",
       "      <td>LASC002598,LASC010177,LASC010178,LASC010181,LA...</td>\n",
       "      <td>LASC005289,LASC005293,LASC007879,LASC007882,LA...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>101-W7_S13</td>\n",
       "      <td>101.0</td>\n",
       "      <td>DRR296821</td>\n",
       "      <td>pilipes</td>\n",
       "      <td>Nephila</td>\n",
       "      <td>Araneidae</td>\n",
       "      <td>8.071637352461016</td>\n",
       "      <td>8.646158335980173</td>\n",
       "      <td>9.98759718640447</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26.0</td>\n",
       "      <td>39.6</td>\n",
       "      <td>10.4</td>\n",
       "      <td>2.75</td>\n",
       "      <td>0.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>282</td>\n",
       "      <td>7416-W1_S13</td>\n",
       "      <td>7416.0</td>\n",
       "      <td>DRR168635</td>\n",
       "      <td>-</td>\n",
       "      <td>Cyrtophora</td>\n",
       "      <td>Araneidae</td>\n",
       "      <td>6.685198909399794</td>\n",
       "      <td>5.3871184863428105</td>\n",
       "      <td>8.271033619828959</td>\n",
       "      <td>...</td>\n",
       "      <td>293.0</td>\n",
       "      <td>317.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>54.3</td>\n",
       "      <td>2.7</td>\n",
       "      <td>2.49</td>\n",
       "      <td>0.06</td>\n",
       "      <td>3.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>283</td>\n",
       "      <td>7421-W1_S15</td>\n",
       "      <td>7421.0</td>\n",
       "      <td>DRR168623</td>\n",
       "      <td>extrusa</td>\n",
       "      <td>Caerostris</td>\n",
       "      <td>Araneidae</td>\n",
       "      <td>7.256716041040335</td>\n",
       "      <td>7.211023294093093</td>\n",
       "      <td>9.12808312070015</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23.0</td>\n",
       "      <td>45.3</td>\n",
       "      <td>2.1</td>\n",
       "      <td>1.45</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.2</td>\n",
       "      <td>11.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>284</td>\n",
       "      <td>7426-W1_S6</td>\n",
       "      <td>7426.0</td>\n",
       "      <td>DRR168628</td>\n",
       "      <td>-</td>\n",
       "      <td>Acrosomoides</td>\n",
       "      <td>Araneidae</td>\n",
       "      <td>4.737890606089994</td>\n",
       "      <td>3.446384610800192</td>\n",
       "      <td>7.49681745895526</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.0</td>\n",
       "      <td>33.6</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.23</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39.3</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>285</td>\n",
       "      <td>7439-W1_S16</td>\n",
       "      <td>7439.0</td>\n",
       "      <td>DRR168624</td>\n",
       "      <td>darwini</td>\n",
       "      <td>Caerostris</td>\n",
       "      <td>Araneidae</td>\n",
       "      <td>8.635579390054453</td>\n",
       "      <td>7.468948571055397</td>\n",
       "      <td>9.36551228527911</td>\n",
       "      <td>...</td>\n",
       "      <td>268.0</td>\n",
       "      <td>290.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>42.5</td>\n",
       "      <td>1.2</td>\n",
       "      <td>3.44</td>\n",
       "      <td>0.03</td>\n",
       "      <td>7.1</td>\n",
       "      <td>43.9</td>\n",
       "      <td>4.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>286</td>\n",
       "      <td>7441-W1_S4</td>\n",
       "      <td>7441.0</td>\n",
       "      <td>DRR168617</td>\n",
       "      <td>inaurata</td>\n",
       "      <td>Trichonephila</td>\n",
       "      <td>Araneidae</td>\n",
       "      <td>8.249533259747007</td>\n",
       "      <td>9.38537818630248</td>\n",
       "      <td>9.042138939047234</td>\n",
       "      <td>...</td>\n",
       "      <td>285.0</td>\n",
       "      <td>304.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>47.5</td>\n",
       "      <td>9.7</td>\n",
       "      <td>6.69</td>\n",
       "      <td>0.63</td>\n",
       "      <td>5.8</td>\n",
       "      <td>43.3</td>\n",
       "      <td>3.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>139 rows × 209 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Orthogroup           ID  idv_id        DRR species_x          Genus  \\\n",
       "0        Group          NaN     NaN        NaN       NaN            NaN   \n",
       "1     geneName          NaN     NaN        NaN       NaN            NaN   \n",
       "2      product          NaN     NaN        NaN       NaN            NaN   \n",
       "3       lascID          NaN     NaN        NaN       NaN            NaN   \n",
       "4            0   101-W7_S13   101.0  DRR296821   pilipes        Nephila   \n",
       "..         ...          ...     ...        ...       ...            ...   \n",
       "134        282  7416-W1_S13  7416.0  DRR168635         -     Cyrtophora   \n",
       "135        283  7421-W1_S15  7421.0  DRR168623   extrusa     Caerostris   \n",
       "136        284   7426-W1_S6  7426.0  DRR168628         -   Acrosomoides   \n",
       "137        285  7439-W1_S16  7439.0  DRR168624   darwini     Caerostris   \n",
       "138        286   7441-W1_S4  7441.0  DRR168617  inaurata  Trichonephila   \n",
       "\n",
       "        Family                                          OG0000013  \\\n",
       "0          NaN                                                NaN   \n",
       "1          NaN  cyp4c3_3,cyp4v2_cyp4c3_5,cyp4v2_23,cyp4v2_3,cy...   \n",
       "2          NaN  Cytochrome P450 4c3,Cytochrome P450 4V2,Cytoch...   \n",
       "3          NaN  LASC003001,LASC003994,LASC003995,LASC005732,LA...   \n",
       "4    Araneidae                                  8.071637352461016   \n",
       "..         ...                                                ...   \n",
       "134  Araneidae                                  6.685198909399794   \n",
       "135  Araneidae                                  7.256716041040335   \n",
       "136  Araneidae                                  4.737890606089994   \n",
       "137  Araneidae                                  8.635579390054453   \n",
       "138  Araneidae                                  8.249533259747007   \n",
       "\n",
       "                                             OG0000022  \\\n",
       "0                                                  NaN   \n",
       "1    LASC002598,LASC010177,LASC010178,LASC010181,LA...   \n",
       "2                                 hypothetical protein   \n",
       "3    LASC002598,LASC010177,LASC010178,LASC010181,LA...   \n",
       "4                                    8.646158335980173   \n",
       "..                                                 ...   \n",
       "134                                 5.3871184863428105   \n",
       "135                                  7.211023294093093   \n",
       "136                                  3.446384610800192   \n",
       "137                                  7.468948571055397   \n",
       "138                                   9.38537818630248   \n",
       "\n",
       "                                             OG0000031  ...  \\\n",
       "0                                                  NaN  ...   \n",
       "1    jhamt_16,LASC005293,jhamt_8,jhamt_11,jhamt_5,j...  ...   \n",
       "2    Juvenile hormone acid O-methyltransferase,hypo...  ...   \n",
       "3    LASC005289,LASC005293,LASC007879,LASC007882,LA...  ...   \n",
       "4                                     9.98759718640447  ...   \n",
       "..                                                 ...  ...   \n",
       "134                                  8.271033619828959  ...   \n",
       "135                                   9.12808312070015  ...   \n",
       "136                                   7.49681745895526  ...   \n",
       "137                                   9.36551228527911  ...   \n",
       "138                                  9.042138939047234  ...   \n",
       "\n",
       "    five_percent_weightloss ten_percent_weightloss crystallinity  \\\n",
       "0                       NaN                    NaN           NaN   \n",
       "1                       NaN                    NaN           NaN   \n",
       "2                       NaN                    NaN           NaN   \n",
       "3                       NaN                    NaN           NaN   \n",
       "4                       NaN                    NaN          26.0   \n",
       "..                      ...                    ...           ...   \n",
       "134                   293.0                  317.0          28.0   \n",
       "135                     NaN                    NaN          23.0   \n",
       "136                     NaN                    NaN          25.0   \n",
       "137                   268.0                  290.0          31.0   \n",
       "138                   285.0                  304.0          24.0   \n",
       "\n",
       "    birefringence birefringence_sd diameter diameter_sd water_content  \\\n",
       "0             NaN              NaN      NaN         NaN           NaN   \n",
       "1             NaN              NaN      NaN         NaN           NaN   \n",
       "2             NaN              NaN      NaN         NaN           NaN   \n",
       "3             NaN              NaN      NaN         NaN           NaN   \n",
       "4            39.6             10.4     2.75        0.09           NaN   \n",
       "..            ...              ...      ...         ...           ...   \n",
       "134          54.3              2.7     2.49        0.06           3.6   \n",
       "135          45.3              2.1     1.45        0.04           NaN   \n",
       "136          33.6              2.0     2.23        0.04           NaN   \n",
       "137          42.5              1.2     3.44        0.03           7.1   \n",
       "138          47.5              9.7     6.69        0.63           5.8   \n",
       "\n",
       "    supercontraction supercontraction_sd  \n",
       "0                NaN                 NaN  \n",
       "1                NaN                 NaN  \n",
       "2                NaN                 NaN  \n",
       "3                NaN                 NaN  \n",
       "4                NaN                 NaN  \n",
       "..               ...                 ...  \n",
       "134              NaN                 NaN  \n",
       "135             35.2                11.1  \n",
       "136             39.3                 5.3  \n",
       "137             43.9                 4.4  \n",
       "138             43.3                 3.6  \n",
       "\n",
       "[139 rows x 209 columns]"
      ]
     },
     "execution_count": 279,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mag"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
